{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyP95ftq5OnDthcgzqkR8mjF",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/fogg-lab/transcriptomics-data-query-and-retrieval/blob/main/notebooks/GEO_data_retrieval_demo.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Setup"
      ],
      "metadata": {
        "id": "An0JBNNyhXAe"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 1) Install the Python package"
      ],
      "metadata": {
        "id": "8dTEWErUxRU6"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "FKvKsFFGPgzQ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6002bc09-cb6e-4fbe-c378-293c10c0fbd3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'transcriptomics-data-query-and-retrieval'...\n",
            "remote: Enumerating objects: 266, done.\u001b[K\n",
            "remote: Counting objects: 100% (35/35), done.\u001b[K\n",
            "remote: Compressing objects: 100% (16/16), done.\u001b[K\n",
            "remote: Total 266 (delta 21), reused 28 (delta 19), pack-reused 231\u001b[K\n",
            "Receiving objects: 100% (266/266), 74.89 KiB | 3.12 MiB/s, done.\n",
            "Resolving deltas: 100% (141/141), done.\n",
            "Processing ./transcriptomics-data-query-and-retrieval\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting biopython (from transcriptomics-data-query==0.1)\n",
            "  Downloading biopython-1.81-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m7.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting GEOparse (from transcriptomics-data-query==0.1)\n",
            "  Downloading GEOparse-2.0.3.tar.gz (278 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m278.5/278.5 kB\u001b[0m \u001b[31m13.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from transcriptomics-data-query==0.1) (1.5.3)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transcriptomics-data-query==0.1) (2.31.0)\n",
            "Collecting mygene (from transcriptomics-data-query==0.1)\n",
            "  Downloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython->transcriptomics-data-query==0.1) (1.23.5)\n",
            "Requirement already satisfied: tqdm>=4.31.1 in /usr/local/lib/python3.10/dist-packages (from GEOparse->transcriptomics-data-query==0.1) (4.66.0)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.10/dist-packages (from pandas->transcriptomics-data-query==0.1) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->transcriptomics-data-query==0.1) (2023.3)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transcriptomics-data-query==0.1) (3.2.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transcriptomics-data-query==0.1) (3.4)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transcriptomics-data-query==0.1) (2.0.4)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transcriptomics-data-query==0.1) (2023.7.22)\n",
            "Collecting biothings-client>=0.2.6 (from mygene->transcriptomics-data-query==0.1)\n",
            "  Downloading biothings_client-0.3.0-py2.py3-none-any.whl (29 kB)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.1->pandas->transcriptomics-data-query==0.1) (1.16.0)\n",
            "Building wheels for collected packages: transcriptomics-data-query, GEOparse\n",
            "  Building wheel for transcriptomics-data-query (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for transcriptomics-data-query: filename=transcriptomics_data_query-0.1-py3-none-any.whl size=13110 sha256=98a4044fb93ba22cc430fe7130f43187824c2773ab4d7d8a0246ef6dc846c27d\n",
            "  Stored in directory: /root/.cache/pip/wheels/f6/04/ae/09175569e39ca781736d5cdc0b77543fc1629b610009352108\n",
            "  Building wheel for GEOparse (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for GEOparse: filename=GEOparse-2.0.3-py3-none-any.whl size=29046 sha256=32edf92bec254ef52da2ae4499d8d917a3fff59f38616aef5e23f82cf535c350\n",
            "  Stored in directory: /root/.cache/pip/wheels/14/96/7a/2ccb3a2a9e6f57b1c758b4ced8445fad36bba011bcebb2775f\n",
            "Successfully built transcriptomics-data-query GEOparse\n",
            "Installing collected packages: biopython, biothings-client, mygene, GEOparse, transcriptomics-data-query\n",
            "Successfully installed GEOparse-2.0.3 biopython-1.81 biothings-client-0.3.0 mygene-3.2.2 transcriptomics-data-query-0.1\n"
          ]
        }
      ],
      "source": [
        "# Clone repository\n",
        "!git clone https://github.com/fogg-lab/transcriptomics-data-query-and-retrieval.git\n",
        "\n",
        "# Install package\n",
        "!pip install ./transcriptomics-data-query-and-retrieval"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 2) Install R packages\n",
        "*This will take up to 10 minutes*"
      ],
      "metadata": {
        "id": "-golMCqhxVqb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Extra preliminary steps to install packages faster in Colab/Ubuntu, using bspm\n",
        "!sudo add-apt-repository -y ppa:marutter/rrutter4.0\n",
        "!sudo add-apt-repository -y ppa:c2d4u.team/c2d4u4.0+\n",
        "!sudo apt-get update && sudo apt-get install -y python3-{dbus,gi,apt}\n",
        "!wget https://github.com/Enchufa2/bspm/archive/refs/tags/v0.5.4.tar.gz\n",
        "!sudo R CMD INSTALL v0.5.4.tar.gz\n",
        "!echo \"bspm::enable()\" | sudo tee -a /etc/R/Rprofile.site"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DwMWoDZAI77W",
        "outputId": "ebf40735-fa42-4554-f793-2cf661d88a41"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Repository: 'deb https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu/ jammy main'\n",
            "Description:\n",
            "A PPA for the base R package, from version 4.0 and higher\n",
            "More info: https://launchpad.net/~marutter/+archive/ubuntu/rrutter4.0\n",
            "Adding repository.\n",
            "Adding deb entry to /etc/apt/sources.list.d/marutter-ubuntu-rrutter4_0-jammy.list\n",
            "Adding disabled deb-src entry to /etc/apt/sources.list.d/marutter-ubuntu-rrutter4_0-jammy.list\n",
            "Adding key to /etc/apt/trusted.gpg.d/marutter-ubuntu-rrutter4_0.gpg with fingerprint C9A7585B49D51698710F3A115E25F516B04C661B\n",
            "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "Hit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Get:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]\n",
            "Get:6 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease [18.1 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Get:9 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [832 kB]\n",
            "Get:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease [24.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [49.8 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [918 kB]\n",
            "Get:13 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [979 kB]\n",
            "Get:14 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [857 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1,129 kB]\n",
            "Get:16 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy InRelease [17.5 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,236 kB]\n",
            "Hit:18 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:19 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main Sources [2,165 kB]\n",
            "Get:20 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 Packages [1,112 kB]\n",
            "Get:21 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy/main amd64 Packages [36.0 kB]\n",
            "Get:22 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy/main amd64 Packages [7,237 B]\n",
            "Fetched 9,722 kB in 6s (1,730 kB/s)\n",
            "Reading package lists... Done\n",
            "Repository: 'deb https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu/ jammy main'\n",
            "Description:\n",
            "A PPA for R packages from CRAN's Task Views built against R 4.0 (and subsequent releases).  Only building packages for LTS releases.\n",
            "More info: https://launchpad.net/~c2d4u.team/+archive/ubuntu/c2d4u4.0+\n",
            "Adding repository.\n",
            "Found existing deb entry in /etc/apt/sources.list.d/c2d4u_team-ubuntu-c2d4u4_0_-jammy.list\n",
            "Adding deb entry to /etc/apt/sources.list.d/c2d4u_team-ubuntu-c2d4u4_0_-jammy.list\n",
            "Adding disabled deb-src entry to /etc/apt/sources.list.d/c2d4u_team-ubuntu-c2d4u4_0_-jammy.list\n",
            "Adding key to /etc/apt/trusted.gpg.d/c2d4u_team-ubuntu-c2d4u4_0_.gpg with fingerprint 6E12762B81063D17BDDD3142F142A4D99F16EB04\n",
            "Hit:1 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit:6 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
            "Hit:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy InRelease\n",
            "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Reading package lists... Done\n",
            "Hit:1 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:7 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy InRelease\n",
            "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Reading package lists... Done\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "python3-dbus is already the newest version (1.2.18-3build1).\n",
            "python3-dbus set to manually installed.\n",
            "python3-apt is already the newest version (2.4.0ubuntu2).\n",
            "python3-apt set to manually installed.\n",
            "python3-gi is already the newest version (3.42.1-0ubuntu1).\n",
            "python3-gi set to manually installed.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 23 not upgraded.\n",
            "--2023-08-15 01:49:44--  https://github.com/Enchufa2/bspm/archive/refs/tags/v0.5.4.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.113.3\n",
            "Connecting to github.com (github.com)|140.82.113.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://codeload.github.com/Enchufa2/bspm/tar.gz/refs/tags/v0.5.4 [following]\n",
            "--2023-08-15 01:49:44--  https://codeload.github.com/Enchufa2/bspm/tar.gz/refs/tags/v0.5.4\n",
            "Resolving codeload.github.com (codeload.github.com)... 140.82.112.9\n",
            "Connecting to codeload.github.com (codeload.github.com)|140.82.112.9|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [application/x-gzip]\n",
            "Saving to: ‘v0.5.4.tar.gz’\n",
            "\n",
            "v0.5.4.tar.gz           [  <=>               ]   2.74M  12.2MB/s    in 0.2s    \n",
            "\n",
            "2023-08-15 01:49:44 (12.2 MB/s) - ‘v0.5.4.tar.gz’ saved [2869637]\n",
            "\n",
            "Warning in untar2(tarfile, files, list, exdir, restore_times) :\n",
            "  skipping pax global extended headers\n",
            "* installing to library ‘/usr/local/lib/R/site-library’\n",
            "* installing *source* package ‘bspm’ ...\n",
            "** using staged installation\n",
            "* installing /usr/share/dbus-1/system-services/org.r_project.linux1.service\n",
            "* installing /etc/dbus-1/system.d/org.r_project.linux1.conf\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "** testing if installed package can be loaded from temporary location\n",
            "** testing if installed package can be loaded from final location\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (bspm)\n",
            "bspm::enable()\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Install packages using the script, install_r_packages.R\n",
        "!sudo Rscript ./transcriptomics-data-query-and-retrieval/install_r_packages.R\n",
        "\n",
        "# Install preprocessCore in single threaded mode to avoid threading bug on Colab\n",
        "!git clone https://github.com/bmbolstad/preprocessCore.git\n",
        "!R CMD INSTALL --configure-args=\"--disable-threading\" ./preprocessCore"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mM6-KJaiqlHQ",
        "outputId": "68e5cc78-a23e-4b3a-dc15-abdda574c202"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "Hit http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
            "Hit https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy InRelease\n",
            "Hit https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Fetched 0 B in 0s (0 B/s)\n",
            "Available system packages as root...\n",
            "Install system packages as root...\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Reading package lists... Done    \n",
            "Building dependency tree... Done \n",
            "Reading state information... Done\n",
            "Get:1 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-biocmanager all 1.30.22-1cran1.2204.0 [479 kB]\n",
            "Fetched 479 kB in 0s (0 B/s)\n",
            "Selecting previously unselected package r-cran-biocmanager.\n",
            "(Reading database ... 120828 files and directories currently installed.)\n",
            "Preparing to unpack .../r-cran-biocmanager_1.30.22-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-biocmanager (1.30.22-1cran1.2204.0) ...\n",
            "Setting up r-cran-biocmanager (1.30.22-1cran1.2204.0) ...\n",
            "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
            "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
            "Replacement repositories:\n",
            "    CRAN: https://cran.rstudio.com\n",
            "Bioconductor version 3.17 (BiocManager 1.30.22), R 4.3.1 (2023-06-16)\n",
            "Installing package(s) 'BiocVersion'\n",
            "Available system packages as root...\n",
            "\n",
            "  There is a binary version available but the source version is later:\n",
            "            binary source\n",
            "BiocVersion 3.14.0 3.17.1\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/BiocVersion_3.17.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 971 bytes\n",
            "==================================================\n",
            "downloaded 971 bytes\n",
            "\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘BiocVersion’ ...\n",
            "** using staged installation\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (BiocVersion)\n",
            "\n",
            "The downloaded source packages are in\n",
            "\t‘/tmp/RtmpNjbuEp/downloaded_packages’\n",
            "Old packages: 'bit', 'bslib', 'cpp11', 'curl', 'devtools', 'gargle', 'gert',\n",
            "  'ggplot2', 'highr', 'htmltools', 'isoband', 'openssl', 'purrr', 'rmarkdown',\n",
            "  'roxygen2', 'tinytex', 'whisker', 'xfun', 'boot', 'foreign', 'Matrix',\n",
            "  'nlme', 'survival'\n",
            "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
            "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
            "Replacement repositories:\n",
            "    CRAN: https://cran.rstudio.com\n",
            "Bioconductor version 3.17 (BiocManager 1.30.22), R 4.3.1 (2023-06-16)\n",
            "Installing package(s) 'edgeR', 'sva', 'oligo', 'pd.clariom.d.human',\n",
            "  'pd.hg.u133.plus.2'\n",
            "Available system packages as root...\n",
            "\n",
            "  There are binary versions available but the source versions are later:\n",
            "             binary source\n",
            "png           0.1-7  0.1-8\n",
            "oligoClasses 1.56.0 1.62.0\n",
            "affxparser   1.66.0 1.72.0\n",
            "S4Arrays      1.0.4  1.0.5\n",
            "DelayedArray 0.26.6 0.26.7\n",
            "oligo        1.58.0 1.64.1\n",
            "\n",
            "Install system packages as root...\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Reading package lists... Done    \n",
            "Building dependency tree... Done \n",
            "Reading state information... Done\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-plogr all 0.2.0-3build1 [15.1 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-xtable all 1:1.8-4-2 [689 kB]\n",
            "Get:3 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-zlibbioc amd64 1.46.0-1cran1.2204.0 [15.3 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-lambda.r all 1.2.4-2build1 [119 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-futile.options all 1.0.1-3build1 [20.2 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-futile.logger all 1.4.3-4build1 [99.3 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 r-cran-snow all 1:0.4.4-2 [97.6 kB]\n",
            "Get:8 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-affyio amd64 1.70.0-1cran1.2204.0 [89.3 kB]\n",
            "Get:9 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-biocgenerics all 0.46.0-1cran1.2204.0 [589 kB]\n",
            "Get:10 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-biobase amd64 2.60.0-1cran1.2204.0 [2149 kB]\n",
            "Get:11 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-s4vectors amd64 0.38.1-1cran1.2204.0 [2253 kB]\n",
            "Get:12 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-iranges amd64 2.34.1-1cran1.2204.0 [2249 kB]\n",
            "Get:13 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-rsqlite amd64 2.3.1-1cran1.2204.0 [1167 kB]\n",
            "Get:14 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-png amd64 0.1-7-4cran2.2204.0 [41.6 kB]\n",
            "Get:15 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-xvector amd64 0.40.0-1cran1.2204.0 [623 kB]\n",
            "Get:16 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-bitops amd64 1.0-7-1cran1.2204.0 [27.3 kB]\n",
            "Get:17 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-rcurl amd64 1.98-1.12-1cran1.2204.0 [803 kB]\n",
            "Get:18 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-genomeinfodbdata all 1.2.10-1cran1.2204.0 [12.0 MB]\n",
            "Get:19 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-genomeinfodb all 1.36.1-1cran1.2204.0 [4087 kB]\n",
            "Get:20 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-biostrings amd64 2.68.1-1cran1.2204.0 [13.7 MB]\n",
            "Get:21 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-keggrest all 1.40.0-1cran1.2204.0 [186 kB]\n",
            "Get:22 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-annotationdbi all 1.62.2-1cran1.2204.0 [4951 kB]\n",
            "Get:23 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-xml amd64 3.99-0.14-1cran1.2204.0 [1711 kB]\n",
            "Get:24 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-annotate all 1.78.0-1cran1.2204.0 [1726 kB]\n",
            "Get:25 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-formatr all 1.14-1cran1.2204.0 [152 kB]\n",
            "Get:26 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-bh all 1.81.0-1-1cran1.2204.0 [8776 kB]\n",
            "Get:27 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-biocparallel amd64 1.34.2-1cran1.2204.0 [1354 kB]\n",
            "Get:28 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-matrixstats amd64 1.0.0-1cran1.2204.0 [434 kB]\n",
            "Get:29 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-matrixgenerics all 1.12.3-1cran1.2204.0 [408 kB]\n",
            "Get:30 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-s4arrays amd64 1.0.4-1cran1.2204.0 [692 kB]\n",
            "Get:31 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-delayedarray amd64 0.26.6-1cran1.2204.0 [1972 kB]\n",
            "Get:32 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-limma amd64 3.56.2-1cran1.2204.0 [2799 kB]\n",
            "Get:33 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-locfit amd64 1.5-9.8-1cran1.2204.0 [514 kB]\n",
            "Get:34 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-rcpp amd64 1.0.11-1cran1.2204.0 [1978 kB]\n",
            "Get:35 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-edger amd64 3.42.4-1cran1.2204.0 [2635 kB]\n",
            "Get:36 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-genefilter amd64 1.82.1-1cran1.2204.0 [1129 kB]\n",
            "Get:37 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-genomicranges amd64 1.52.0-1cran1.2204.0 [2031 kB]\n",
            "Get:38 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-preprocesscore amd64 1.62.1-1cran1.2204.0 [149 kB]\n",
            "Get:39 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-summarizedexperiment all 1.30.2-1cran1.2204.0 [1443 kB]\n",
            "Get:40 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-bioc-sva amd64 3.48.0-1cran1.2204.0 [466 kB]\n",
            "Get:41 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-abind all 1.4-5-2.1ubuntu1.2204.0 [64.1 kB]\n",
            "Get:42 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-ff amd64 4.0.9-1cran1.2204.0 [960 kB]\n",
            "Get:43 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-iterators all 1.0.14-1cran1.2204.0 [335 kB]\n",
            "Get:44 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-foreach all 1.5.2-1cran1.2204.0 [124 kB]\n",
            "Get:45 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 r-cran-testit all 0.13-1cran1.2204.0 [32.8 kB]\n",
            "Fetched 77.9 MB in 6s (1389 kB/s)\n",
            "Selecting previously unselected package r-bioc-zlibbioc.\n",
            "(Reading database ... 120862 files and directories currently installed.)\n",
            "Preparing to unpack .../00-r-bioc-zlibbioc_1.46.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-zlibbioc (1.46.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-affyio.\n",
            "Preparing to unpack .../01-r-bioc-affyio_1.70.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-affyio (1.70.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-biocgenerics.\n",
            "Preparing to unpack .../02-r-bioc-biocgenerics_0.46.0-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-biocgenerics (0.46.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-biobase.\n",
            "Preparing to unpack .../03-r-bioc-biobase_2.60.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-biobase (2.60.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-s4vectors.\n",
            "Preparing to unpack .../04-r-bioc-s4vectors_0.38.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-s4vectors (0.38.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-iranges.\n",
            "Preparing to unpack .../05-r-bioc-iranges_2.34.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-iranges (2.34.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-plogr.\n",
            "Preparing to unpack .../06-r-cran-plogr_0.2.0-3build1_all.deb ...\n",
            "Unpacking r-cran-plogr (0.2.0-3build1) ...\n",
            "Selecting previously unselected package r-cran-rsqlite.\n",
            "Preparing to unpack .../07-r-cran-rsqlite_2.3.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-rsqlite (2.3.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-png.\n",
            "Preparing to unpack .../08-r-cran-png_0.1-7-4cran2.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-png (0.1-7-4cran2.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-xvector.\n",
            "Preparing to unpack .../09-r-bioc-xvector_0.40.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-xvector (0.40.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-bitops.\n",
            "Preparing to unpack .../10-r-cran-bitops_1.0-7-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-bitops (1.0-7-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-rcurl.\n",
            "Preparing to unpack .../11-r-cran-rcurl_1.98-1.12-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-rcurl (1.98-1.12-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-genomeinfodbdata.\n",
            "Preparing to unpack .../12-r-bioc-genomeinfodbdata_1.2.10-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-genomeinfodbdata (1.2.10-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-genomeinfodb.\n",
            "Preparing to unpack .../13-r-bioc-genomeinfodb_1.36.1-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-genomeinfodb (1.36.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-biostrings.\n",
            "Preparing to unpack .../14-r-bioc-biostrings_2.68.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-biostrings (2.68.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-keggrest.\n",
            "Preparing to unpack .../15-r-bioc-keggrest_1.40.0-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-keggrest (1.40.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-annotationdbi.\n",
            "Preparing to unpack .../16-r-bioc-annotationdbi_1.62.2-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-annotationdbi (1.62.2-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-xml.\n",
            "Preparing to unpack .../17-r-cran-xml_3.99-0.14-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-xml (3.99-0.14-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-xtable.\n",
            "Preparing to unpack .../18-r-cran-xtable_1%3a1.8-4-2_all.deb ...\n",
            "Unpacking r-cran-xtable (1:1.8-4-2) ...\n",
            "Selecting previously unselected package r-bioc-annotate.\n",
            "Preparing to unpack .../19-r-bioc-annotate_1.78.0-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-annotate (1.78.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-formatr.\n",
            "Preparing to unpack .../20-r-cran-formatr_1.14-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-formatr (1.14-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-lambda.r.\n",
            "Preparing to unpack .../21-r-cran-lambda.r_1.2.4-2build1_all.deb ...\n",
            "Unpacking r-cran-lambda.r (1.2.4-2build1) ...\n",
            "Selecting previously unselected package r-cran-futile.options.\n",
            "Preparing to unpack .../22-r-cran-futile.options_1.0.1-3build1_all.deb ...\n",
            "Unpacking r-cran-futile.options (1.0.1-3build1) ...\n",
            "Selecting previously unselected package r-cran-futile.logger.\n",
            "Preparing to unpack .../23-r-cran-futile.logger_1.4.3-4build1_all.deb ...\n",
            "Unpacking r-cran-futile.logger (1.4.3-4build1) ...\n",
            "Selecting previously unselected package r-cran-snow.\n",
            "Preparing to unpack .../24-r-cran-snow_1%3a0.4.4-2_all.deb ...\n",
            "Unpacking r-cran-snow (1:0.4.4-2) ...\n",
            "Selecting previously unselected package r-cran-bh.\n",
            "Preparing to unpack .../25-r-cran-bh_1.81.0-1-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-bh (1.81.0-1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-biocparallel.\n",
            "Preparing to unpack .../26-r-bioc-biocparallel_1.34.2-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-biocparallel (1.34.2-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-matrixstats.\n",
            "Preparing to unpack .../27-r-cran-matrixstats_1.0.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-matrixstats (1.0.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-matrixgenerics.\n",
            "Preparing to unpack .../28-r-bioc-matrixgenerics_1.12.3-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-matrixgenerics (1.12.3-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-s4arrays.\n",
            "Preparing to unpack .../29-r-bioc-s4arrays_1.0.4-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-s4arrays (1.0.4-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-delayedarray.\n",
            "Preparing to unpack .../30-r-bioc-delayedarray_0.26.6-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-delayedarray (0.26.6-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-limma.\n",
            "Preparing to unpack .../31-r-bioc-limma_3.56.2-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-limma (3.56.2-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-locfit.\n",
            "Preparing to unpack .../32-r-cran-locfit_1.5-9.8-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-locfit (1.5-9.8-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-rcpp.\n",
            "Preparing to unpack .../33-r-cran-rcpp_1.0.11-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-rcpp (1.0.11-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-edger.\n",
            "Preparing to unpack .../34-r-bioc-edger_3.42.4-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-edger (3.42.4-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-genefilter.\n",
            "Preparing to unpack .../35-r-bioc-genefilter_1.82.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-genefilter (1.82.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-genomicranges.\n",
            "Preparing to unpack .../36-r-bioc-genomicranges_1.52.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-genomicranges (1.52.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-preprocesscore.\n",
            "Preparing to unpack .../37-r-bioc-preprocesscore_1.62.1-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-preprocesscore (1.62.1-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-summarizedexperiment.\n",
            "Preparing to unpack .../38-r-bioc-summarizedexperiment_1.30.2-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-bioc-summarizedexperiment (1.30.2-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-bioc-sva.\n",
            "Preparing to unpack .../39-r-bioc-sva_3.48.0-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-bioc-sva (3.48.0-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-abind.\n",
            "Preparing to unpack .../40-r-cran-abind_1.4-5-2.1ubuntu1.2204.0_all.deb ...\n",
            "Unpacking r-cran-abind (1.4-5-2.1ubuntu1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-ff.\n",
            "Preparing to unpack .../41-r-cran-ff_4.0.9-1cran1.2204.0_amd64.deb ...\n",
            "Unpacking r-cran-ff (4.0.9-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-iterators.\n",
            "Preparing to unpack .../42-r-cran-iterators_1.0.14-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-iterators (1.0.14-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-foreach.\n",
            "Preparing to unpack .../43-r-cran-foreach_1.5.2-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-foreach (1.5.2-1cran1.2204.0) ...\n",
            "Selecting previously unselected package r-cran-testit.\n",
            "Preparing to unpack .../44-r-cran-testit_0.13-1cran1.2204.0_all.deb ...\n",
            "Unpacking r-cran-testit (0.13-1cran1.2204.0) ...\n",
            "Setting up r-cran-bh (1.81.0-1-1cran1.2204.0) ...\n",
            "Setting up r-cran-rcpp (1.0.11-1cran1.2204.0) ...\n",
            "Setting up r-bioc-biocgenerics (0.46.0-1cran1.2204.0) ...\n",
            "Setting up r-cran-matrixstats (1.0.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-matrixgenerics (1.12.3-1cran1.2204.0) ...\n",
            "Setting up r-cran-plogr (0.2.0-3build1) ...\n",
            "Setting up r-cran-iterators (1.0.14-1cran1.2204.0) ...\n",
            "Setting up r-cran-abind (1.4-5-2.1ubuntu1.2204.0) ...\n",
            "Setting up r-cran-foreach (1.5.2-1cran1.2204.0) ...\n",
            "Setting up r-cran-snow (1:0.4.4-2) ...\n",
            "Setting up r-cran-ff (4.0.9-1cran1.2204.0) ...\n",
            "Setting up r-cran-xtable (1:1.8-4-2) ...\n",
            "Setting up r-cran-bitops (1.0-7-1cran1.2204.0) ...\n",
            "Setting up r-bioc-zlibbioc (1.46.0-1cran1.2204.0) ...\n",
            "Setting up r-cran-png (0.1-7-4cran2.2204.0) ...\n",
            "Setting up r-bioc-biobase (2.60.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-s4vectors (0.38.1-1cran1.2204.0) ...\n",
            "Setting up r-bioc-limma (3.56.2-1cran1.2204.0) ...\n",
            "Setting up r-cran-futile.options (1.0.1-3build1) ...\n",
            "Setting up r-cran-formatr (1.14-1cran1.2204.0) ...\n",
            "Setting up r-cran-xml (3.99-0.14-1cran1.2204.0) ...\n",
            "Setting up r-cran-locfit (1.5-9.8-1cran1.2204.0) ...\n",
            "Setting up r-bioc-preprocesscore (1.62.1-1cran1.2204.0) ...\n",
            "Setting up r-bioc-iranges (2.34.1-1cran1.2204.0) ...\n",
            "Setting up r-cran-rsqlite (2.3.1-1cran1.2204.0) ...\n",
            "Setting up r-cran-rcurl (1.98-1.12-1cran1.2204.0) ...\n",
            "Setting up r-cran-testit (0.13-1cran1.2204.0) ...\n",
            "Setting up r-bioc-genomeinfodbdata (1.2.10-1cran1.2204.0) ...\n",
            "Setting up r-bioc-affyio (1.70.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-s4arrays (1.0.4-1cran1.2204.0) ...\n",
            "Setting up r-bioc-edger (3.42.4-1cran1.2204.0) ...\n",
            "Setting up r-cran-lambda.r (1.2.4-2build1) ...\n",
            "Setting up r-bioc-genomeinfodb (1.36.1-1cran1.2204.0) ...\n",
            "Setting up r-cran-futile.logger (1.4.3-4build1) ...\n",
            "Setting up r-bioc-xvector (0.40.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-delayedarray (0.26.6-1cran1.2204.0) ...\n",
            "Setting up r-bioc-biocparallel (1.34.2-1cran1.2204.0) ...\n",
            "Setting up r-bioc-genomicranges (1.52.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-biostrings (2.68.1-1cran1.2204.0) ...\n",
            "Setting up r-bioc-summarizedexperiment (1.30.2-1cran1.2204.0) ...\n",
            "Setting up r-bioc-keggrest (1.40.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-annotationdbi (1.62.2-1cran1.2204.0) ...\n",
            "Setting up r-bioc-annotate (1.78.0-1cran1.2204.0) ...\n",
            "Setting up r-bioc-genefilter (1.82.1-1cran1.2204.0) ...\n",
            "Setting up r-bioc-sva (3.48.0-1cran1.2204.0) ...\n",
            "trying URL 'https://cran.rstudio.com/src/contrib/png_0.1-8.tar.gz'\n",
            "Content type 'application/x-gzip' length 24880 bytes (24 KB)\n",
            "==================================================\n",
            "downloaded 24 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/oligoClasses_1.62.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 373301 bytes (364 KB)\n",
            "==================================================\n",
            "downloaded 364 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/affxparser_1.72.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 801513 bytes (782 KB)\n",
            "==================================================\n",
            "downloaded 782 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/S4Arrays_1.0.5.tar.gz'\n",
            "Content type 'application/x-gzip' length 279242 bytes (272 KB)\n",
            "==================================================\n",
            "downloaded 272 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/DelayedArray_0.26.7.tar.gz'\n",
            "Content type 'application/x-gzip' length 628192 bytes (613 KB)\n",
            "==================================================\n",
            "downloaded 613 KB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/bioc/src/contrib/oligo_1.64.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 28421223 bytes (27.1 MB)\n",
            "==================================================\n",
            "downloaded 27.1 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/data/annotation/src/contrib/pd.clariom.d.human_3.14.1.tar.gz'\n",
            "Content type 'application/x-gzip' length 415158722 bytes (395.9 MB)\n",
            "==================================================\n",
            "downloaded 395.9 MB\n",
            "\n",
            "trying URL 'https://bioconductor.org/packages/3.17/data/annotation/src/contrib/pd.hg.u133.plus.2_3.12.0.tar.gz'\n",
            "Content type 'application/x-gzip' length 24799467 bytes (23.7 MB)\n",
            "==================================================\n",
            "downloaded 23.7 MB\n",
            "\n",
            "begin installing package ‘png’\n",
            "begin installing package ‘S4Arrays’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘png’ ...\n",
            "** package ‘png’ successfully unpacked and MD5 sums checked\n",
            "** using staged installation\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "make[1]: Entering directory '/tmp/Rtmpy3m6rA/R.INSTALL1ac0609f67ad/png/src'\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG       `libpng-config --cflags` -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dummy.c -o dummy.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG       `libpng-config --cflags` -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c read.c -o read.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG       `libpng-config --cflags` -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c write.c -o write.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o png.so dummy.o read.o write.o -lpng16 -lm -lz -lm -L/usr/lib/R/lib -lR\n",
            "make[1]: Leaving directory '/tmp/Rtmpy3m6rA/R.INSTALL1ac0609f67ad/png/src'\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-png/00new/png/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (png)\n",
            "begin installing package ‘affxparser’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘S4Arrays’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "make[1]: Entering directory '/tmp/Rtmpcw9UAn/R.INSTALL1ac21edff89b/S4Arrays/src'\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_S4Arrays.c -o R_init_S4Arrays.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c abind.c -o abind.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c array_selection.c -o array_selection.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dim_tuning_utils.c -o dim_tuning_utils.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o S4Arrays.so R_init_S4Arrays.o S4Vectors_stubs.o abind.o array_selection.o dim_tuning_utils.o -L/usr/lib/R/lib -lR\n",
            "make[1]: Leaving directory '/tmp/Rtmpcw9UAn/R.INSTALL1ac21edff89b/S4Arrays/src'\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-S4Arrays/00new/S4Arrays/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "Creating a new generic function for ‘rowsum’ in package ‘S4Arrays’\n",
            "Creating a new generic function for ‘abind’ in package ‘S4Arrays’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (S4Arrays)\n",
            "begin installing package ‘DelayedArray’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘DelayedArray’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "make[1]: Entering directory '/tmp/RtmpR2q0dJ/R.INSTALL1c7d579b33ff/DelayedArray/src'\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_DelayedArray.c -o R_init_DelayedArray.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c compress_atomic_vector.c -o compress_atomic_vector.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sparseMatrix_utils.c -o sparseMatrix_utils.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o DelayedArray.so R_init_DelayedArray.o S4Vectors_stubs.o compress_atomic_vector.o sparseMatrix_utils.o -L/usr/lib/R/lib -lR\n",
            "make[1]: Leaving directory '/tmp/RtmpR2q0dJ/R.INSTALL1c7d579b33ff/DelayedArray/src'\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-DelayedArray/00new/DelayedArray/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "Creating a new generic function for ‘apply’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘sweep’ in package ‘DelayedArray’\n",
            "Creating a new generic function for ‘scale’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘pnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qnorm’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘pbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qbinom’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dpois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘ppois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qpois’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘dlogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘plogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "Creating a generic function for ‘qlogis’ from package ‘stats’ in package ‘DelayedArray’\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (DelayedArray)\n",
            "begin installing package ‘oligoClasses’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘oligoClasses’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (oligoClasses)\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘affxparser’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "using C++ compiler: ‘g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "make[1]: Entering directory '/tmp/Rtmpv3CD1S/R.INSTALL1b60208bff2f/affxparser/src'\n",
            "rm -f fusion/calvin_files/data/src/CDFData.o fusion/calvin_files/data/src/CDFProbeGroupInformation.o fusion/calvin_files/data/src/CDFProbeInformation.o fusion/calvin_files/data/src/CDFProbeSetInformation.o fusion/calvin_files/data/src/CDFQCProbeInformation.o fusion/calvin_files/data/src/CDFQCProbeSetInformation.o fusion/calvin_files/data/src/CELData.o fusion/calvin_files/data/src/CHPBackgroundZone.o fusion/calvin_files/data/src/CHPData.o fusion/calvin_files/data/src/CHPExpressionEntry.o fusion/calvin_files/data/src/CHPMultiDataData.o fusion/calvin_files/data/src/CHPTilingData.o fusion/calvin_files/data/src/CHPQuantificationData.o fusion/calvin_files/data/src/CHPQuantificationDetectionData.o fusion/calvin_files/data/src/CHPGenotypeEntry.o fusion/calvin_files/data/src/CHPUniversalEntry.o fusion/calvin_files/data/src/ColumnInfo.o fusion/calvin_files/data/src/DataGroup.o fusion/calvin_files/data/src/DataGroupHeader.o fusion/calvin_files/data/src/DataSet.o fusion/calvin_files/data/src/DataSetHeader.o fusion/calvin_files/data/src/FileHeader.o fusion/calvin_files/data/src/GenericData.o fusion/calvin_files/data/src/GenericDataHeader.o fusion/calvin_files/exception/src/ExceptionBase.o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCELDataAdapter.o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCHPDataAdapter.o fusion/calvin_files/fusion/src/FusionBPMAPData.o fusion/calvin_files/fusion/src/FusionCDFData.o fusion/calvin_files/fusion/src/FusionCDFQCProbeSetNames.o fusion/calvin_files/fusion/src/FusionCELData.o fusion/calvin_files/fusion/src/FusionCHPData.o fusion/calvin_files/fusion/src/FusionProbeSetResults.o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCELDataAdapter.o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCHPDataAdapter.o fusion/calvin_files/fusion/src/FusionCHPLegacyData.o fusion/calvin_files/fusion/src/FusionCHPMultiDataAccessor.o fusion/calvin_files/fusion/src/FusionCHPMultiDataData.o fusion/calvin_files/fusion/src/FusionCHPTilingData.o fusion/calvin_files/fusion/src/FusionCHPGenericData.o fusion/calvin_files/fusion/src/FusionCHPQuantificationData.o fusion/calvin_files/fusion/src/FusionCHPQuantificationDetectionData.o fusion/calvin_files/parameter/src/ParameterNameValueType.o fusion/calvin_files/parsers/src/CDFFileReader.o fusion/calvin_files/parsers/src/CelFileReader.o fusion/calvin_files/parsers/src/CHPFileReader.o fusion/calvin_files/parsers/src/CHPMultiDataFileReader.o fusion/calvin_files/parsers/src/CHPTilingFileReader.o fusion/calvin_files/parsers/src/CHPQuantificationFileReader.o fusion/calvin_files/parsers/src/CHPQuantificationDetectionFileReader.o fusion/calvin_files/parsers/src/DataGroupHeaderReader.o fusion/calvin_files/parsers/src/DataGroupReader.o fusion/calvin_files/parsers/src/DataSetHeaderReader.o fusion/calvin_files/parsers/src/DataSetReader.o fusion/calvin_files/parsers/src/FileHeaderReader.o fusion/calvin_files/parsers/src/FileInput.o fusion/calvin_files/parsers/src/GenericDataHeaderReader.o fusion/calvin_files/parsers/src/GenericFileReader.o fusion/calvin_files/utils/src/AffymetrixGuid.o fusion/calvin_files/utils/src/DateTime.o fusion/calvin_files/utils/src/FileUtils.o fusion/calvin_files/utils/src/StringUtils.o fusion/calvin_files/utils/src/checksum.o fusion/file/BPMAPFileData.o fusion/file/BPMAPFileWriter.o fusion/file/CDFFileData.o fusion/file/CELFileData.o fusion/file/CHPFileData.o fusion/file/FileIO.o fusion/file/FileWriter.o fusion/file/TsvFile/ClfFile.o fusion/file/TsvFile/PgfFile.o fusion/file/TsvFile/TsvFile.o fusion/util/AffxByteArray.o fusion/util/AffxConv.o fusion/util/MsgStream.o fusion/util/Util.o fusion/util/Err.o fusion/util/Fs.o fusion/util/Verbose.o fusion/util/RowFile.o fusion/util/TableFile.o fusion/util/Convert.o R_affx_cel_parser.o R_affx_cdf_parser.o R_affx_cdf_extras.o R_affx_bpmap_parser.o R_affx_clf_pgf_parser.o R_affx_chp_parser.o 000.init.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFData.cpp -o fusion/calvin_files/data/src/CDFData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFProbeGroupInformation.cpp -o fusion/calvin_files/data/src/CDFProbeGroupInformation.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFProbeInformation.cpp -o fusion/calvin_files/data/src/CDFProbeInformation.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFProbeSetInformation.cpp -o fusion/calvin_files/data/src/CDFProbeSetInformation.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFQCProbeInformation.cpp -o fusion/calvin_files/data/src/CDFQCProbeInformation.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CDFQCProbeSetInformation.cpp -o fusion/calvin_files/data/src/CDFQCProbeSetInformation.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CELData.cpp -o fusion/calvin_files/data/src/CELData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPBackgroundZone.cpp -o fusion/calvin_files/data/src/CHPBackgroundZone.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPData.cpp -o fusion/calvin_files/data/src/CHPData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPExpressionEntry.cpp -o fusion/calvin_files/data/src/CHPExpressionEntry.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPMultiDataData.cpp -o fusion/calvin_files/data/src/CHPMultiDataData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPTilingData.cpp -o fusion/calvin_files/data/src/CHPTilingData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPQuantificationData.cpp -o fusion/calvin_files/data/src/CHPQuantificationData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPQuantificationDetectionData.cpp -o fusion/calvin_files/data/src/CHPQuantificationDetectionData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPGenotypeEntry.cpp -o fusion/calvin_files/data/src/CHPGenotypeEntry.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/CHPUniversalEntry.cpp -o fusion/calvin_files/data/src/CHPUniversalEntry.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/ColumnInfo.cpp -o fusion/calvin_files/data/src/ColumnInfo.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/DataGroup.cpp -o fusion/calvin_files/data/src/DataGroup.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/DataGroupHeader.cpp -o fusion/calvin_files/data/src/DataGroupHeader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/DataSet.cpp -o fusion/calvin_files/data/src/DataSet.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/DataSetHeader.cpp -o fusion/calvin_files/data/src/DataSetHeader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/FileHeader.cpp -o fusion/calvin_files/data/src/FileHeader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/GenericData.cpp -o fusion/calvin_files/data/src/GenericData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/data/src/GenericDataHeader.cpp -o fusion/calvin_files/data/src/GenericDataHeader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/exception/src/ExceptionBase.cpp -o fusion/calvin_files/exception/src/ExceptionBase.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCELDataAdapter.cpp -o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCELDataAdapter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCHPDataAdapter.cpp -o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCHPDataAdapter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionBPMAPData.cpp -o fusion/calvin_files/fusion/src/FusionBPMAPData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCDFData.cpp -o fusion/calvin_files/fusion/src/FusionCDFData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCDFQCProbeSetNames.cpp -o fusion/calvin_files/fusion/src/FusionCDFQCProbeSetNames.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCELData.cpp -o fusion/calvin_files/fusion/src/FusionCELData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPData.cpp -o fusion/calvin_files/fusion/src/FusionCHPData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionProbeSetResults.cpp -o fusion/calvin_files/fusion/src/FusionProbeSetResults.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCELDataAdapter.cpp -o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCELDataAdapter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCHPDataAdapter.cpp -o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCHPDataAdapter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPLegacyData.cpp -o fusion/calvin_files/fusion/src/FusionCHPLegacyData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPMultiDataAccessor.cpp -o fusion/calvin_files/fusion/src/FusionCHPMultiDataAccessor.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPMultiDataData.cpp -o fusion/calvin_files/fusion/src/FusionCHPMultiDataData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPTilingData.cpp -o fusion/calvin_files/fusion/src/FusionCHPTilingData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPGenericData.cpp -o fusion/calvin_files/fusion/src/FusionCHPGenericData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPQuantificationData.cpp -o fusion/calvin_files/fusion/src/FusionCHPQuantificationData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/fusion/src/FusionCHPQuantificationDetectionData.cpp -o fusion/calvin_files/fusion/src/FusionCHPQuantificationDetectionData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parameter/src/ParameterNameValueType.cpp -o fusion/calvin_files/parameter/src/ParameterNameValueType.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CDFFileReader.cpp -o fusion/calvin_files/parsers/src/CDFFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CelFileReader.cpp -o fusion/calvin_files/parsers/src/CelFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CHPFileReader.cpp -o fusion/calvin_files/parsers/src/CHPFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CHPMultiDataFileReader.cpp -o fusion/calvin_files/parsers/src/CHPMultiDataFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CHPTilingFileReader.cpp -o fusion/calvin_files/parsers/src/CHPTilingFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CHPQuantificationFileReader.cpp -o fusion/calvin_files/parsers/src/CHPQuantificationFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/CHPQuantificationDetectionFileReader.cpp -o fusion/calvin_files/parsers/src/CHPQuantificationDetectionFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/DataGroupHeaderReader.cpp -o fusion/calvin_files/parsers/src/DataGroupHeaderReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/DataGroupReader.cpp -o fusion/calvin_files/parsers/src/DataGroupReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/DataSetHeaderReader.cpp -o fusion/calvin_files/parsers/src/DataSetHeaderReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/DataSetReader.cpp -o fusion/calvin_files/parsers/src/DataSetReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/FileHeaderReader.cpp -o fusion/calvin_files/parsers/src/FileHeaderReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/FileInput.cpp -o fusion/calvin_files/parsers/src/FileInput.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/GenericDataHeaderReader.cpp -o fusion/calvin_files/parsers/src/GenericDataHeaderReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/parsers/src/GenericFileReader.cpp -o fusion/calvin_files/parsers/src/GenericFileReader.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/utils/src/AffymetrixGuid.cpp -o fusion/calvin_files/utils/src/AffymetrixGuid.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/utils/src/DateTime.cpp -o fusion/calvin_files/utils/src/DateTime.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/utils/src/FileUtils.cpp -o fusion/calvin_files/utils/src/FileUtils.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/utils/src/StringUtils.cpp -o fusion/calvin_files/utils/src/StringUtils.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/calvin_files/utils/src/checksum.cpp -o fusion/calvin_files/utils/src/checksum.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/BPMAPFileData.cpp -o fusion/file/BPMAPFileData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/BPMAPFileWriter.cpp -o fusion/file/BPMAPFileWriter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/CDFFileData.cpp -o fusion/file/CDFFileData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/CELFileData.cpp -o fusion/file/CELFileData.o\n",
            "fusion/file/CELFileData.cpp: In member function ‘float affxcel::CCELFileData::GetIntensity(int)’:\n",
            "fusion/file/CELFileData.cpp:2409:41: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2409 |                 fIntensity=MmGetFloat_I(&m_pEntries[index].Intensity);\n",
            "      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2414:41: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2414 |                 fIntensity=MmGetFloat_I(&(m_pEntries[index].Intensity));\n",
            "      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2419:42: warning: taking address of packed member of ‘affxcel::_CELFileTranscriptomeEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2419 |                 fIntensity=MmGetUInt16_N(&(m_pTransciptomeEntries[index].Intensity));\n",
            "      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘int affxcel::CCELFileData::GetIntensities(int, std::vector<float>&)’:\n",
            "fusion/file/CELFileData.cpp:2445:37: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2445 |       intensities[idx]=MmGetFloat_I(&m_pEntries[idx].Intensity);\n",
            "      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2451:37: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2451 |       intensities[idx]=MmGetFloat_I(&m_pEntries[idx].Intensity);\n",
            "      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2457:38: warning: taking address of packed member of ‘affxcel::_CELFileTranscriptomeEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2457 |       intensities[idx]=MmGetUInt16_N(&m_pTransciptomeEntries[idx].Intensity);\n",
            "      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘float affxcel::CCELFileData::GetStdv(int)’:\n",
            "fusion/file/CELFileData.cpp:2506:37: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2506 |                 fStdev=MmGetFloat_I(&m_pEntries[index].Stdv);\n",
            "      |                                     ^~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2509:37: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2509 |                 fStdev=MmGetFloat_I(&(m_pEntries[index].Stdv));\n",
            "      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2512:38: warning: taking address of packed member of ‘affxcel::_CELFileTranscriptomeEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2512 |                 fStdev=MmGetUInt16_N(&(m_pTransciptomeEntries[index].Stdv));\n",
            "      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘short int affxcel::CCELFileData::GetPixels(int)’:\n",
            "fusion/file/CELFileData.cpp:2551:38: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2551 |                 sPixels=MmGetInt16_I(&m_pEntries[index].Pixels);\n",
            "      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:2554:38: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 2554 |                 sPixels=MmGetInt16_I(&m_pEntries[index].Pixels);\n",
            "      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘void affxcel::CCELFileData::SetIntensity(int, float)’:\n",
            "fusion/file/CELFileData.cpp:3194:30: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3194 |                 MmSetFloat_I(&m_pEntries[index].Intensity,intensity);\n",
            "      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:3199:30: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3199 |                 MmSetFloat_I(&m_pEntries[index].Intensity,intensity);\n",
            "      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:3204:31: warning: taking address of packed member of ‘affxcel::_CELFileTranscriptomeEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3204 |                 MmSetUInt16_N(&m_pTransciptomeEntries[index].Intensity,s);\n",
            "      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘void affxcel::CCELFileData::SetStdv(int, float)’:\n",
            "fusion/file/CELFileData.cpp:3251:31: warning: taking address of packed member of ‘affxcel::_CELFileTranscriptomeEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3251 |                 MmSetUInt16_N(&m_pTransciptomeEntries[index].Stdv,s);\n",
            "      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp:3255:30: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3255 |                 MmSetFloat_I(&m_pEntries[index].Stdv,stdev);\n",
            "      |                              ^~~~~~~~~~~~~~~~~~~~~~~\n",
            "fusion/file/CELFileData.cpp: In member function ‘void affxcel::CCELFileData::SetPixels(int, short int)’:\n",
            "fusion/file/CELFileData.cpp:3294:42: warning: taking address of packed member of ‘affxcel::_CELFileEntryType’ may result in an unaligned pointer value [-Waddress-of-packed-member]\n",
            " 3294 |                 MmSetUInt16_I((uint16_t*)&m_pEntries[index].Pixels,(uint16_t)pixels);\n",
            "      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/CHPFileData.cpp -o fusion/file/CHPFileData.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/FileIO.cpp -o fusion/file/FileIO.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/FileWriter.cpp -o fusion/file/FileWriter.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/TsvFile/ClfFile.cpp -o fusion/file/TsvFile/ClfFile.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/TsvFile/PgfFile.cpp -o fusion/file/TsvFile/PgfFile.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/file/TsvFile/TsvFile.cpp -o fusion/file/TsvFile/TsvFile.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/AffxByteArray.cpp -o fusion/util/AffxByteArray.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/AffxConv.cpp -o fusion/util/AffxConv.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/MsgStream.cpp -o fusion/util/MsgStream.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/Util.cpp -o fusion/util/Util.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/Err.cpp -o fusion/util/Err.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/Fs.cpp -o fusion/util/Fs.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/Verbose.cpp -o fusion/util/Verbose.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/RowFile.cpp -o fusion/util/RowFile.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/TableFile.cpp -o fusion/util/TableFile.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c fusion/util/Convert.cpp -o fusion/util/Convert.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_cel_parser.cpp -o R_affx_cel_parser.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_cdf_parser.cpp -o R_affx_cdf_parser.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_cdf_extras.cpp -o R_affx_cdf_extras.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_bpmap_parser.cpp -o R_affx_bpmap_parser.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_clf_pgf_parser.cpp -o R_affx_clf_pgf_parser.o\n",
            "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -Wno-sign-compare -O0 -c R_affx_chp_parser.cpp -o R_affx_chp_parser.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG -I. -Ifusion/calvin_files/array/src -Ifusion/calvin_files/data/src -Ifusion/calvin_files/exception/src -Ifusion/calvin_files/fusion/src -Ifusion/calvin_files/fusion/src/GCOSAdapter -Ifusion/calvin_files/fusion/src/CalvinAdapter -Ifusion/calvin_files/parameter/src -Ifusion/calvin_files/parsers/src -Ifusion/calvin_files/portability/src -Ifusion/calvin_files/template/src -Ifusion/calvin_files/utils/src -Ifusion/calvin_files/writers/src -Ifusion/file -Ifusion/file/TsvFile -Ifusion/portability -Ifusion/util -Ifusion -D_USE_MEM_MAPPING_      -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 000.init.c -o 000.init.o\n",
            "g++ -std=gnu++17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o affxparser.so fusion/calvin_files/data/src/CDFData.o fusion/calvin_files/data/src/CDFProbeGroupInformation.o fusion/calvin_files/data/src/CDFProbeInformation.o fusion/calvin_files/data/src/CDFProbeSetInformation.o fusion/calvin_files/data/src/CDFQCProbeInformation.o fusion/calvin_files/data/src/CDFQCProbeSetInformation.o fusion/calvin_files/data/src/CELData.o fusion/calvin_files/data/src/CHPBackgroundZone.o fusion/calvin_files/data/src/CHPData.o fusion/calvin_files/data/src/CHPExpressionEntry.o fusion/calvin_files/data/src/CHPMultiDataData.o fusion/calvin_files/data/src/CHPTilingData.o fusion/calvin_files/data/src/CHPQuantificationData.o fusion/calvin_files/data/src/CHPQuantificationDetectionData.o fusion/calvin_files/data/src/CHPGenotypeEntry.o fusion/calvin_files/data/src/CHPUniversalEntry.o fusion/calvin_files/data/src/ColumnInfo.o fusion/calvin_files/data/src/DataGroup.o fusion/calvin_files/data/src/DataGroupHeader.o fusion/calvin_files/data/src/DataSet.o fusion/calvin_files/data/src/DataSetHeader.o fusion/calvin_files/data/src/FileHeader.o fusion/calvin_files/data/src/GenericData.o fusion/calvin_files/data/src/GenericDataHeader.o fusion/calvin_files/exception/src/ExceptionBase.o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCELDataAdapter.o fusion/calvin_files/fusion/src/CalvinAdapter/CalvinCHPDataAdapter.o fusion/calvin_files/fusion/src/FusionBPMAPData.o fusion/calvin_files/fusion/src/FusionCDFData.o fusion/calvin_files/fusion/src/FusionCDFQCProbeSetNames.o fusion/calvin_files/fusion/src/FusionCELData.o fusion/calvin_files/fusion/src/FusionCHPData.o fusion/calvin_files/fusion/src/FusionProbeSetResults.o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCELDataAdapter.o fusion/calvin_files/fusion/src/GCOSAdapter/GCOSCHPDataAdapter.o fusion/calvin_files/fusion/src/FusionCHPLegacyData.o fusion/calvin_files/fusion/src/FusionCHPMultiDataAccessor.o fusion/calvin_files/fusion/src/FusionCHPMultiDataData.o fusion/calvin_files/fusion/src/FusionCHPTilingData.o fusion/calvin_files/fusion/src/FusionCHPGenericData.o fusion/calvin_files/fusion/src/FusionCHPQuantificationData.o fusion/calvin_files/fusion/src/FusionCHPQuantificationDetectionData.o fusion/calvin_files/parameter/src/ParameterNameValueType.o fusion/calvin_files/parsers/src/CDFFileReader.o fusion/calvin_files/parsers/src/CelFileReader.o fusion/calvin_files/parsers/src/CHPFileReader.o fusion/calvin_files/parsers/src/CHPMultiDataFileReader.o fusion/calvin_files/parsers/src/CHPTilingFileReader.o fusion/calvin_files/parsers/src/CHPQuantificationFileReader.o fusion/calvin_files/parsers/src/CHPQuantificationDetectionFileReader.o fusion/calvin_files/parsers/src/DataGroupHeaderReader.o fusion/calvin_files/parsers/src/DataGroupReader.o fusion/calvin_files/parsers/src/DataSetHeaderReader.o fusion/calvin_files/parsers/src/DataSetReader.o fusion/calvin_files/parsers/src/FileHeaderReader.o fusion/calvin_files/parsers/src/FileInput.o fusion/calvin_files/parsers/src/GenericDataHeaderReader.o fusion/calvin_files/parsers/src/GenericFileReader.o fusion/calvin_files/utils/src/AffymetrixGuid.o fusion/calvin_files/utils/src/DateTime.o fusion/calvin_files/utils/src/FileUtils.o fusion/calvin_files/utils/src/StringUtils.o fusion/calvin_files/utils/src/checksum.o fusion/file/BPMAPFileData.o fusion/file/BPMAPFileWriter.o fusion/file/CDFFileData.o fusion/file/CELFileData.o fusion/file/CHPFileData.o fusion/file/FileIO.o fusion/file/FileWriter.o fusion/file/TsvFile/ClfFile.o fusion/file/TsvFile/PgfFile.o fusion/file/TsvFile/TsvFile.o fusion/util/AffxByteArray.o fusion/util/AffxConv.o fusion/util/MsgStream.o fusion/util/Util.o fusion/util/Err.o fusion/util/Fs.o fusion/util/Verbose.o fusion/util/RowFile.o fusion/util/TableFile.o fusion/util/Convert.o R_affx_cel_parser.o R_affx_cdf_parser.o R_affx_cdf_extras.o R_affx_bpmap_parser.o R_affx_clf_pgf_parser.o R_affx_chp_parser.o 000.init.o -L/usr/lib/R/lib -lR\n",
            "make[1]: Leaving directory '/tmp/Rtmpv3CD1S/R.INSTALL1b60208bff2f/affxparser/src'\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-affxparser/00new/affxparser/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (affxparser)\n",
            "begin installing package ‘oligo’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘oligo’ ...\n",
            "** using staged installation\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "make[1]: Entering directory '/tmp/Rtmpbmd5Hl/R.INSTALL20b06917f801/oligo/src'\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c DABG.c -o DABG.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ParserGzXYS.c -o ParserGzXYS.o\n",
            "ParserGzXYS.c: In function ‘gzcountLines’:\n",
            "ParserGzXYS.c:18:17: warning: passing argument 1 of ‘gzeof’ from incompatible pointer type [-Wincompatible-pointer-types]\n",
            "   18 |   while (!gzeof(file)){\n",
            "      |                 ^~~~\n",
            "      |                 |\n",
            "      |                 struct gzFile_s **\n",
            "In file included from /usr/include/zlib.h:34,\n",
            "                 from ParserGzXYS.c:4:\n",
            "/usr/include/zlib.h:1595:27: note: expected ‘gzFile’ {aka ‘struct gzFile_s *’} but argument is of type ‘struct gzFile_s **’\n",
            " 1595 | ZEXTERN int ZEXPORT gzeof OF((gzFile file));\n",
            "      |                           ^~\n",
            "ParserGzXYS.c:19:12: warning: passing argument 1 of ‘gzgets’ from incompatible pointer type [-Wincompatible-pointer-types]\n",
            "   19 |     gzgets(file, buffer, 1000);\n",
            "      |            ^~~~\n",
            "      |            |\n",
            "      |            struct gzFile_s **\n",
            "In file included from /usr/include/zlib.h:34,\n",
            "                 from ParserGzXYS.c:4:\n",
            "/usr/include/zlib.h:1492:31: note: expected ‘gzFile’ {aka ‘struct gzFile_s *’} but argument is of type ‘struct gzFile_s **’\n",
            " 1492 | ZEXTERN char * ZEXPORT gzgets OF((gzFile file, char *buf, int len));\n",
            "      |                               ^~\n",
            "ParserGzXYS.c: In function ‘gzxys_header_field’:\n",
            "ParserGzXYS.c:46:6: warning: assignment to ‘struct gzFile_s **’ from incompatible pointer type ‘gzFile’ {aka ‘struct gzFile_s *’} [-Wincompatible-pointer-types]\n",
            "   46 |   fp = gzopen(currentFile, \"rb\");\n",
            "      |      ^\n",
            "ParserGzXYS.c:50:10: warning: passing argument 1 of ‘gzgets’ from incompatible pointer type [-Wincompatible-pointer-types]\n",
            "   50 |   gzgets(fp, buffer, LINEMAX);\n",
            "      |          ^~\n",
            "      |          |\n",
            "      |          struct gzFile_s **\n",
            "In file included from /usr/include/zlib.h:34,\n",
            "                 from ParserGzXYS.c:4:\n",
            "/usr/include/zlib.h:1492:31: note: expected ‘gzFile’ {aka ‘struct gzFile_s *’} but argument is of type ‘struct gzFile_s **’\n",
            " 1492 | ZEXTERN char * ZEXPORT gzgets OF((gzFile file, char *buf, int len));\n",
            "      |                               ^~\n",
            "ParserGzXYS.c:51:11: warning: passing argument 1 of ‘gzclose’ from incompatible pointer type [-Wincompatible-pointer-types]\n",
            "   51 |   gzclose(fp);\n",
            "      |           ^~\n",
            "      |           |\n",
            "      |           struct gzFile_s **\n",
            "In file included from /usr/include/zlib.h:34,\n",
            "                 from ParserGzXYS.c:4:\n",
            "/usr/include/zlib.h:1631:32: note: expected ‘gzFile’ {aka ‘struct gzFile_s *’} but argument is of type ‘struct gzFile_s **’\n",
            " 1631 | ZEXTERN int ZEXPORT    gzclose OF((gzFile file));\n",
            "      |                                ^~\n",
            "ParserGzXYS.c: In function ‘R_read_gzxys_files’:\n",
            "ParserGzXYS.c:134:6: warning: assignment to ‘struct gzFile_s **’ from incompatible pointer type ‘gzFile’ {aka ‘struct gzFile_s *’} [-Wincompatible-pointer-types]\n",
            "  134 |   fp = gzopen(CHAR(STRING_ELT(filenames, 0)), \"rb\");\n",
            "      |      ^\n",
            "ParserGzXYS.c:138:11: warning: passing argument 1 of ‘gzclose’ from incompatible pointer type [-Wincompatible-pointer-types]\n",
            "  138 |   gzclose(fp);\n",
            "      |           ^~\n",
            "      |           |\n",
            "      |           struct gzFile_s **\n",
            "In file included from /usr/include/zlib.h:34,\n",
            "                 from ParserGzXYS.c:4:\n",
            "/usr/include/zlib.h:1631:32: note: expected ‘gzFile’ {aka ‘struct gzFile_s *’} but argument is of type ‘struct gzFile_s **’\n",
            " 1631 | ZEXTERN int ZEXPORT    gzclose OF((gzFile file));\n",
            "      |                                ^~\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ParserXYS.c -o ParserXYS.o\n",
            "ParserXYS.c: In function ‘xys_header_field’:\n",
            "ParserXYS.c:145:3: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]\n",
            "  145 |   fgets(buffer, LINEMAX, fp);\n",
            "      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "ParserXYS.c: In function ‘R_read_xys_header’:\n",
            "ParserXYS.c:320:3: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]\n",
            "  320 |   fgets(buffer, LINEMAX, fp);\n",
            "      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c baseProfile.c -o baseProfile.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c basecontent.c -o basecontent.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c chipbackground.c -o chipbackground.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c mas5calls.c -o mas5calls.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rma2.c -o rma2.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rma_common.c -o rma_common.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/preprocessCore/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c trimmed.c -o trimmed.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o oligo.so DABG.o ParserGzXYS.o ParserXYS.o baseProfile.o basecontent.o chipbackground.o mas5calls.o rma2.o rma_common.o trimmed.o -lz -lgfortran -lm -lquadmath -llapack -lblas -L/usr/lib/R/lib -lR\n",
            "make[1]: Leaving directory '/tmp/Rtmpbmd5Hl/R.INSTALL20b06917f801/oligo/src'\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-oligo/00new/oligo/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "Note: in method for ‘pm<-’ with signature\n",
            "‘object=\"TilingFeatureSet\",subset=\"ANY\",target=\"ANY\",value=\"array\"’:\n",
            "expanding the signature to include omitted arguments in definition:\n",
            "subset = \"missing\", target = \"missing\"\n",
            "Note: in method for ‘mm<-’ with signature\n",
            "‘object=\"TilingFeatureSet\",subset=\"ANY\",target=\"ANY\",value=\"array\"’:\n",
            "expanding the signature to include omitted arguments in definition:\n",
            "subset = \"missing\", target = \"missing\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** installing vignettes\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (oligo)\n",
            "begin installing package ‘pd.clariom.d.human’\n",
            "begin installing package ‘pd.hg.u133.plus.2’\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘pd.hg.u133.plus.2’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (pd.hg.u133.plus.2)\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing *source* package ‘pd.clariom.d.human’ ...\n",
            "** using staged installation\n",
            "** R\n",
            "** data\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (pd.clariom.d.human)\n",
            "\n",
            "The downloaded source packages are in\n",
            "\t‘/tmp/RtmpNjbuEp/downloaded_packages’\n",
            "Old packages: 'bit', 'bslib', 'cpp11', 'curl', 'DelayedArray', 'devtools',\n",
            "  'gargle', 'gert', 'ggplot2', 'highr', 'htmltools', 'isoband', 'openssl',\n",
            "  'png', 'purrr', 'rmarkdown', 'roxygen2', 'S4Arrays', 'tinytex', 'whisker',\n",
            "  'xfun', 'boot', 'foreign', 'Matrix', 'nlme', 'survival'\n",
            "Cloning into 'preprocessCore'...\n",
            "remote: Enumerating objects: 1338, done.\u001b[K\n",
            "remote: Counting objects: 100% (93/93), done.\u001b[K\n",
            "remote: Compressing objects: 100% (64/64), done.\u001b[K\n",
            "remote: Total 1338 (delta 53), reused 63 (delta 29), pack-reused 1245\u001b[K\n",
            "Receiving objects: 100% (1338/1338), 408.63 KiB | 7.43 MiB/s, done.\n",
            "Resolving deltas: 100% (926/926), done.\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "* installing to library ‘/usr/local/lib/R/site-library’\n",
            "* installing *source* package ‘preprocessCore’ ...\n",
            "** using staged installation\n",
            "'config' variable 'CPP' is defunct\n",
            "checking for gcc... gcc\n",
            "checking whether the C compiler works... yes\n",
            "checking for C compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether we are using the GNU C compiler... yes\n",
            "checking whether gcc accepts -g... yes\n",
            "checking for gcc option to accept ISO C89... none needed\n",
            "checking how to run the C preprocessor... gcc -E\n",
            "checking for library containing pthread_create... none required\n",
            "checking for grep that handles long lines and -e... /usr/bin/grep\n",
            "checking for egrep... /usr/bin/grep -E\n",
            "checking for ANSI C header files... yes\n",
            "checking for sys/types.h... yes\n",
            "checking for sys/stat.h... yes\n",
            "checking for stdlib.h... yes\n",
            "checking for string.h... yes\n",
            "checking for memory.h... yes\n",
            "checking for strings.h... yes\n",
            "checking for inttypes.h... yes\n",
            "checking for stdint.h... yes\n",
            "checking for unistd.h... yes\n",
            "checking for stdlib.h... (cached) yes\n",
            "checking if PTHREAD_STACK_MIN is defined... yes\n",
            "checking if R is using flexiblas... flexiblas not found. preprocessCore threading will not be disabled\n",
            "configure: Disabling threading for preprocessCore\n",
            "configure: creating ./config.status\n",
            "config.status: creating src/Makevars\n",
            "** libs\n",
            "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_colSummarize.c -o R_colSummarize.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_plmd_interfaces.c -o R_plmd_interfaces.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_plmr_interfaces.c -o R_plmr_interfaces.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_rlm_interfaces.c -o R_rlm_interfaces.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_subColSummarize.c -o R_subColSummarize.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_subrcModel_interfaces.c -o R_subrcModel_interfaces.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c avg.c -o avg.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c avg_log.c -o avg_log.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c biweight.c -o biweight.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c init_package.c -o init_package.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lm.c -o lm.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c log_avg.c -o log_avg.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c log_median.c -o log_median.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c matrix_functions.c -o matrix_functions.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c median.c -o median.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c median_log.c -o median_log.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c medianpolish.c -o medianpolish.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c plmd.c -o plmd.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c plmr.c -o plmr.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c psi_fns.c -o psi_fns.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c qnorm.c -o qnorm.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rlm.c -o rlm.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rlm_anova.c -o rlm_anova.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rlm_se.c -o rlm_se.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rma_background4.c -o rma_background4.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rma_common.c -o rma_common.o\n",
            "gcc -I\"/usr/share/R/include\" -DNDEBUG      -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -DPACKAGE_NAME=\\\"\\\" -DPACKAGE_TARNAME=\\\"\\\" -DPACKAGE_VERSION=\\\"\\\" -DPACKAGE_STRING=\\\"\\\" -DPACKAGE_BUGREPORT=\\\"\\\" -DPACKAGE_URL=\\\"\\\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -fpic  -g -O2 -ffile-prefix-map=/build/r-base-MHXHhT/r-base-4.3.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c weightedkerneldensity.c -o weightedkerneldensity.o\n",
            "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o preprocessCore.so R_colSummarize.o R_plmd_interfaces.o R_plmr_interfaces.o R_rlm_interfaces.o R_subColSummarize.o R_subrcModel_interfaces.o avg.o avg_log.o biweight.o init_package.o lm.o log_avg.o log_median.o matrix_functions.o median.o median_log.o medianpolish.o plmd.o plmr.o psi_fns.o qnorm.o rlm.o rlm_anova.o rlm_se.o rma_background4.o rma_common.o weightedkerneldensity.o -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n",
            "installing to /usr/local/lib/R/site-library/00LOCK-preprocessCore/00new/preprocessCore/libs\n",
            "** R\n",
            "** inst\n",
            "** byte-compile and prepare package for lazy loading\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** help\n",
            "*** installing help indices\n",
            "** building package indices\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package can be loaded from temporary location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** checking absolute paths in shared objects and dynamic libraries\n",
            "** testing if installed package can be loaded from final location\n",
            "Loading required package: utils\n",
            "Tracing function \"install.packages\" in package \"utils\"\n",
            "** testing if installed package keeps a record of temporary installation path\n",
            "* DONE (preprocessCore)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## GEO retrieval and preprocessing example\n",
        "\n",
        "Prepare expression data from a GEO study. This process is similar for microarray and RNASeq data, with some minor differences. Refer to the [documentation](https://github.com/fogg-lab/transcriptomics-data-query-and-retrieval/blob/main/DOCUMENTATION.md) for more information."
      ],
      "metadata": {
        "id": "AX9rKDsicxB-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "### 1. Import packages\n",
        "\n",
        "import GEOparse\n",
        "import pandas as pd\n",
        "import transcriptomics_data_query as tdq"
      ],
      "metadata": {
        "id": "8YbEH102vqDa"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### 2. Obtain a GSE object for a GEO series accession using the GEOparse package\n",
        "\n",
        "# Link: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE161750\n",
        "accession = \"GSE161750\"\n",
        "gse = GEOparse.get_GEO(accession)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "U8zFq90dvwa5",
        "outputId": "ff059017-ae9a-4967-ee64-dda13d8ff449"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "15-Aug-2023 02:51:27 DEBUG utils - Directory ./ already exists. Skipping.\n",
            "DEBUG:GEOparse:Directory ./ already exists. Skipping.\n",
            "15-Aug-2023 02:51:27 INFO GEOparse - File already exist: using local version.\n",
            "INFO:GEOparse:File already exist: using local version.\n",
            "15-Aug-2023 02:51:27 INFO GEOparse - Parsing ./GSE161750_family.soft.gz: \n",
            "INFO:GEOparse:Parsing ./GSE161750_family.soft.gz: \n",
            "15-Aug-2023 02:51:27 DEBUG GEOparse - DATABASE: GeoMiame\n",
            "DEBUG:GEOparse:DATABASE: GeoMiame\n",
            "15-Aug-2023 02:51:27 DEBUG GEOparse - SERIES: GSE161750\n",
            "DEBUG:GEOparse:SERIES: GSE161750\n",
            "15-Aug-2023 02:51:27 DEBUG GEOparse - PLATFORM: GPL23159\n",
            "DEBUG:GEOparse:PLATFORM: GPL23159\n",
            "15-Aug-2023 02:51:30 WARNING GEOTypes - Detected duplicated columns in d GPL23159. Correcting.\n",
            "\n",
            "WARNING:GEOparse:Detected duplicated columns in d GPL23159. Correcting.\n",
            "\n",
            "15-Aug-2023 02:51:30 DEBUG GEOparse - SAMPLE: GSM4913974\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913974\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913975\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913975\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913976\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913976\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913977\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913977\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913978\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913978\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913979\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913979\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913980\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913980\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913981\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913981\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913982\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913982\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913983\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913983\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913984\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913984\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913985\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913985\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913986\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913986\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913987\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913987\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913988\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913988\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913989\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913989\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913990\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913990\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913991\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913991\n",
            "15-Aug-2023 02:51:31 DEBUG GEOparse - SAMPLE: GSM4913992\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913992\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913993\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913993\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913994\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913994\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913995\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913995\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913996\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913996\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913997\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913997\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913998\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913998\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4913999\n",
            "DEBUG:GEOparse:SAMPLE: GSM4913999\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914000\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914000\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914001\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914001\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914002\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914002\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914003\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914003\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914004\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914004\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914005\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914005\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914006\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914006\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914007\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914007\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914008\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914008\n",
            "15-Aug-2023 02:51:32 DEBUG GEOparse - SAMPLE: GSM4914009\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914009\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914010\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914010\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914011\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914011\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914012\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914012\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914013\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914013\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914014\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914014\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914015\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914015\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914016\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914016\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914017\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914017\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914018\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914018\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914019\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914019\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914020\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914020\n",
            "15-Aug-2023 02:51:33 DEBUG GEOparse - SAMPLE: GSM4914021\n",
            "DEBUG:GEOparse:SAMPLE: GSM4914021\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### 3. Download the raw data\n",
        "\n",
        "# If the output_dir argument is not specified, it will be created in the working directory.\n",
        "# For microarray data, this results in a directory of CEL.gz files.\n",
        "# If the accession was an RNASeq study, a counts file would be downloaded instead.\n",
        "tdq.geo.download_geo_expression_data(gse)"
      ],
      "metadata": {
        "id": "xmDPeZeVwoab"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### 4. Download and parse clinical characteristics\n",
        "\n",
        "# Save in current working directory, ./GSE161750_clinical.tsv\n",
        "clinical_file = f\"{accession}_clinical.tsv\"\n",
        "tdq.geo.get_geo_clinical_characteristics(gse, output_file=clinical_file)\n",
        "\n",
        "# View parsed clinical characteristics.\n",
        "# Probably need to clean it up manually in a spreadsheet program.\n",
        "pd.read_csv(clinical_file, sep='\\t')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "DGkpZgAVz3WG",
        "outputId": "62b8df37-a76f-4740-fcc9-843758f505b8"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     sample_id  tissue                             title  \\\n",
              "0   GSM4913974  muscle        Muscle_Baseline_Acute_rep1   \n",
              "1   GSM4913975  muscle        Muscle_Baseline_Acute_rep2   \n",
              "2   GSM4913976  muscle        Muscle_Baseline_Acute_rep3   \n",
              "3   GSM4913977  muscle        Muscle_Baseline_Acute_rep4   \n",
              "4   GSM4913978  muscle        Muscle_Baseline_Acute_rep5   \n",
              "5   GSM4913979  muscle        Muscle_Baseline_Acute_rep6   \n",
              "6   GSM4913980  muscle        Muscle_Baseline_Acute_rep7   \n",
              "7   GSM4913981  muscle        Muscle_Baseline_Acute_rep8   \n",
              "8   GSM4913982  muscle        Muscle_Baseline_Acute_rep9   \n",
              "9   GSM4913983  muscle       Muscle_Baseline_Acute_rep10   \n",
              "10  GSM4913984  muscle       Muscle_Baseline_Acute_rep11   \n",
              "11  GSM4913985  muscle       Muscle_Baseline_Acute_rep12   \n",
              "12  GSM4913986  muscle       Muscle_Baseline_Acute_rep13   \n",
              "13  GSM4913987  muscle       Muscle_Baseline_Acute_rep14   \n",
              "14  GSM4913988  muscle       Muscle_Baseline_Acute_rep15   \n",
              "15  GSM4913989  muscle       Muscle_Baseline_Acute_rep16   \n",
              "16  GSM4913990  muscle       Muscle_Baseline_Acute_rep17   \n",
              "17  GSM4913991  muscle       Muscle_Baseline_Acute_rep18   \n",
              "18  GSM4913992  muscle       Muscle_Baseline_Acute_rep19   \n",
              "19  GSM4913993  muscle       Muscle_Baseline_Acute_rep20   \n",
              "20  GSM4913994  muscle       Muscle_Baseline_Acute_rep21   \n",
              "21  GSM4913995  muscle       Muscle_Baseline_Acute_rep22   \n",
              "22  GSM4913996  muscle       Muscle_Baseline_Acute_rep23   \n",
              "23  GSM4913997  muscle       Muscle_Baseline_Acute_rep24   \n",
              "24  GSM4913998  muscle       Muscle_Baseline_Acute_rep25   \n",
              "25  GSM4913999  muscle   Muscle_Post_Training_Acute_rep1   \n",
              "26  GSM4914000  muscle   Muscle_Post_Training_Acute_rep2   \n",
              "27  GSM4914001  muscle   Muscle_Post_Training_Acute_rep3   \n",
              "28  GSM4914002  muscle   Muscle_Post_Training_Acute_rep4   \n",
              "29  GSM4914003  muscle   Muscle_Post_Training_Acute_rep5   \n",
              "30  GSM4914004  muscle   Muscle_Post_Training_Acute_rep6   \n",
              "31  GSM4914005  muscle   Muscle_Post_Training_Acute_rep7   \n",
              "32  GSM4914006  muscle   Muscle_Post_Training_Acute_rep8   \n",
              "33  GSM4914007  muscle   Muscle_Post_Training_Acute_rep9   \n",
              "34  GSM4914008  muscle  Muscle_Post_Training_Acute_rep10   \n",
              "35  GSM4914009  muscle  Muscle_Post_Training_Acute_rep11   \n",
              "36  GSM4914010  muscle  Muscle_Post_Training_Acute_rep12   \n",
              "37  GSM4914011  muscle  Muscle_Post_Training_Acute_rep13   \n",
              "38  GSM4914012  muscle  Muscle_Post_Training_Acute_rep14   \n",
              "39  GSM4914013  muscle  Muscle_Post_Training_Acute_rep15   \n",
              "40  GSM4914014  muscle  Muscle_Post_Training_Acute_rep16   \n",
              "41  GSM4914015  muscle  Muscle_Post_Training_Acute_rep17   \n",
              "42  GSM4914016  muscle  Muscle_Post_Training_Acute_rep18   \n",
              "43  GSM4914017  muscle  Muscle_Post_Training_Acute_rep19   \n",
              "44  GSM4914018  muscle  Muscle_Post_Training_Acute_rep20   \n",
              "45  GSM4914019  muscle  Muscle_Post_Training_Acute_rep21   \n",
              "46  GSM4914020  muscle  Muscle_Post_Training_Acute_rep22   \n",
              "47  GSM4914021  muscle  Muscle_Post_Training_Acute_rep23   \n",
              "\n",
              "                                          description  \\\n",
              "0   Gene expression data from muscle biopsy untrai...   \n",
              "1   Gene expression data from muscle biopsy untrai...   \n",
              "2   Gene expression data from muscle biopsy untrai...   \n",
              "3   Gene expression data from muscle biopsy untrai...   \n",
              "4   Gene expression data from muscle biopsy untrai...   \n",
              "5   Gene expression data from muscle biopsy untrai...   \n",
              "6   Gene expression data from muscle biopsy untrai...   \n",
              "7   Gene expression data from muscle biopsy untrai...   \n",
              "8   Gene expression data from muscle biopsy untrai...   \n",
              "9   Gene expression data from muscle biopsy untrai...   \n",
              "10  Gene expression data from muscle biopsy untrai...   \n",
              "11  Gene expression data from muscle biopsy untrai...   \n",
              "12  Gene expression data from muscle biopsy untrai...   \n",
              "13  Gene expression data from muscle biopsy untrai...   \n",
              "14  Gene expression data from muscle biopsy untrai...   \n",
              "15  Gene expression data from muscle biopsy untrai...   \n",
              "16  Gene expression data from muscle biopsy untrai...   \n",
              "17  Gene expression data from muscle biopsy untrai...   \n",
              "18  Gene expression data from muscle biopsy untrai...   \n",
              "19  Gene expression data from muscle biopsy untrai...   \n",
              "20  Gene expression data from muscle biopsy untrai...   \n",
              "21  Gene expression data from muscle biopsy untrai...   \n",
              "22  Gene expression data from muscle biopsy untrai...   \n",
              "23  Gene expression data from muscle biopsy untrai...   \n",
              "24  Gene expression data from muscle biopsy untrai...   \n",
              "25  Gene expression data from muscle biopsy traine...   \n",
              "26  Gene expression data from muscle biopsy traine...   \n",
              "27  Gene expression data from muscle biopsy traine...   \n",
              "28  Gene expression data from muscle biopsy traine...   \n",
              "29  Gene expression data from muscle biopsy traine...   \n",
              "30  Gene expression data from muscle biopsy traine...   \n",
              "31  Gene expression data from muscle biopsy traine...   \n",
              "32  Gene expression data from muscle biopsy traine...   \n",
              "33  Gene expression data from muscle biopsy traine...   \n",
              "34  Gene expression data from muscle biopsy traine...   \n",
              "35  Gene expression data from muscle biopsy traine...   \n",
              "36  Gene expression data from muscle biopsy traine...   \n",
              "37  Gene expression data from muscle biopsy traine...   \n",
              "38  Gene expression data from muscle biopsy traine...   \n",
              "39  Gene expression data from muscle biopsy traine...   \n",
              "40  Gene expression data from muscle biopsy traine...   \n",
              "41  Gene expression data from muscle biopsy traine...   \n",
              "42  Gene expression data from muscle biopsy traine...   \n",
              "43  Gene expression data from muscle biopsy traine...   \n",
              "44  Gene expression data from muscle biopsy traine...   \n",
              "45  Gene expression data from muscle biopsy traine...   \n",
              "46  Gene expression data from muscle biopsy traine...   \n",
              "47  Gene expression data from muscle biopsy traine...   \n",
              "\n",
              "                                      source_name_ch1  \n",
              "0   Muscle biopsy untrained after exercise derived...  \n",
              "1   Muscle biopsy untrained after exercise derived...  \n",
              "2   Muscle biopsy untrained after exercise derived...  \n",
              "3   Muscle biopsy untrained after exercise derived...  \n",
              "4   Muscle biopsy untrained after exercise derived...  \n",
              "5   Muscle biopsy untrained after exercise derived...  \n",
              "6   Muscle biopsy untrained after exercise derived...  \n",
              "7   Muscle biopsy untrained after exercise derived...  \n",
              "8   Muscle biopsy untrained after exercise derived...  \n",
              "9   Muscle biopsy untrained after exercise derived...  \n",
              "10  Muscle biopsy untrained after exercise derived...  \n",
              "11  Muscle biopsy untrained after exercise derived...  \n",
              "12  Muscle biopsy untrained after exercise derived...  \n",
              "13  Muscle biopsy untrained after exercise derived...  \n",
              "14  Muscle biopsy untrained after exercise derived...  \n",
              "15  Muscle biopsy untrained after exercise derived...  \n",
              "16  Muscle biopsy untrained after exercise derived...  \n",
              "17  Muscle biopsy untrained after exercise derived...  \n",
              "18  Muscle biopsy untrained after exercise derived...  \n",
              "19  Muscle biopsy untrained after exercise derived...  \n",
              "20  Muscle biopsy untrained after exercise derived...  \n",
              "21  Muscle biopsy untrained after exercise derived...  \n",
              "22  Muscle biopsy untrained after exercise derived...  \n",
              "23  Muscle biopsy untrained after exercise derived...  \n",
              "24  Muscle biopsy untrained after exercise derived...  \n",
              "25  Muscle biopsy trained after exercise derived f...  \n",
              "26  Muscle biopsy trained after exercise derived f...  \n",
              "27  Muscle biopsy trained after exercise derived f...  \n",
              "28  Muscle biopsy trained after exercise derived f...  \n",
              "29  Muscle biopsy trained after exercise derived f...  \n",
              "30  Muscle biopsy trained after exercise derived f...  \n",
              "31  Muscle biopsy trained after exercise derived f...  \n",
              "32  Muscle biopsy trained after exercise derived f...  \n",
              "33  Muscle biopsy trained after exercise derived f...  \n",
              "34  Muscle biopsy trained after exercise derived f...  \n",
              "35  Muscle biopsy trained after exercise derived f...  \n",
              "36  Muscle biopsy trained after exercise derived f...  \n",
              "37  Muscle biopsy trained after exercise derived f...  \n",
              "38  Muscle biopsy trained after exercise derived f...  \n",
              "39  Muscle biopsy trained after exercise derived f...  \n",
              "40  Muscle biopsy trained after exercise derived f...  \n",
              "41  Muscle biopsy trained after exercise derived f...  \n",
              "42  Muscle biopsy trained after exercise derived f...  \n",
              "43  Muscle biopsy trained after exercise derived f...  \n",
              "44  Muscle biopsy trained after exercise derived f...  \n",
              "45  Muscle biopsy trained after exercise derived f...  \n",
              "46  Muscle biopsy trained after exercise derived f...  \n",
              "47  Muscle biopsy trained after exercise derived f...  "
            ],
            "text/html": [
              "\n",
              "\n",
              "  <div id=\"df-50362eba-3d27-4a99-823f-f84b51046690\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>sample_id</th>\n",
              "      <th>tissue</th>\n",
              "      <th>title</th>\n",
              "      <th>description</th>\n",
              "      <th>source_name_ch1</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSM4913974</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep1</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSM4913975</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep2</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSM4913976</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep3</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSM4913977</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep4</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSM4913978</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep5</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>GSM4913979</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep6</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>GSM4913980</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep7</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>GSM4913981</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep8</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>GSM4913982</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep9</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>GSM4913983</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep10</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>GSM4913984</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep11</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>GSM4913985</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep12</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>GSM4913986</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep13</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>GSM4913987</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep14</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>GSM4913988</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep15</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>GSM4913989</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep16</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>GSM4913990</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep17</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>GSM4913991</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep18</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>GSM4913992</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep19</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>GSM4913993</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep20</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>GSM4913994</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep21</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>GSM4913995</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep22</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>GSM4913996</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep23</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>GSM4913997</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep24</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>GSM4913998</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Baseline_Acute_rep25</td>\n",
              "      <td>Gene expression data from muscle biopsy untrai...</td>\n",
              "      <td>Muscle biopsy untrained after exercise derived...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>GSM4913999</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep1</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>GSM4914000</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep2</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>GSM4914001</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep3</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>GSM4914002</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep4</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>GSM4914003</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep5</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>GSM4914004</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep6</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>GSM4914005</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep7</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>GSM4914006</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep8</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>GSM4914007</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep9</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>GSM4914008</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep10</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>GSM4914009</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep11</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>GSM4914010</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep12</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>GSM4914011</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep13</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>GSM4914012</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep14</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>GSM4914013</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep15</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>GSM4914014</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep16</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>GSM4914015</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep17</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>GSM4914016</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep18</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>GSM4914017</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep19</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>GSM4914018</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep20</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>GSM4914019</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep21</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>GSM4914020</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep22</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>GSM4914021</td>\n",
              "      <td>muscle</td>\n",
              "      <td>Muscle_Post_Training_Acute_rep23</td>\n",
              "      <td>Gene expression data from muscle biopsy traine...</td>\n",
              "      <td>Muscle biopsy trained after exercise derived f...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-50362eba-3d27-4a99-823f-f84b51046690')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "\n",
              "\n",
              "\n",
              "    <div id=\"df-63541f5a-d056-46d1-b5d5-f40d056a011a\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-63541f5a-d056-46d1-b5d5-f40d056a011a')\"\n",
              "              title=\"Suggest charts.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "    </div>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "    background-color: #E8F0FE;\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: #1967D2;\n",
              "    height: 32px;\n",
              "    padding: 0 0 0 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: #E2EBFA;\n",
              "    box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: #174EA6;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "    background-color: #3B4455;\n",
              "    fill: #D2E3FC;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart:hover {\n",
              "    background-color: #434B5C;\n",
              "    box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "    filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "    fill: #FFFFFF;\n",
              "  }\n",
              "</style>\n",
              "\n",
              "    <script>\n",
              "      async function quickchart(key) {\n",
              "        const containerElement = document.querySelector('#' + key);\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      }\n",
              "    </script>\n",
              "\n",
              "      <script>\n",
              "\n",
              "function displayQuickchartButton(domScope) {\n",
              "  let quickchartButtonEl =\n",
              "    domScope.querySelector('#df-63541f5a-d056-46d1-b5d5-f40d056a011a button.colab-df-quickchart');\n",
              "  quickchartButtonEl.style.display =\n",
              "    google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "}\n",
              "\n",
              "        displayQuickchartButton(document);\n",
              "      </script>\n",
              "      <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-50362eba-3d27-4a99-823f-f84b51046690 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-50362eba-3d27-4a99-823f-f84b51046690');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 4
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### 5. Normalization (RMA for microarray, TMM for RNASeq)\n",
        "\n",
        "# If the accession is microarray data, input_path should be a directory of CEL.gz files\n",
        "norm_input_path = accession\n",
        "norm_save_path = f\"{accession}_expression_matrix.tsv\"\n",
        "tdq.preprocess.normalize(input_path=norm_input_path, output_file=norm_save_path)\n",
        "\n",
        "# For RNASeq, input_path should be the raw counts file, and clinical_file must also be specified.\n",
        "# norm_input_path = f\"{accession}_expression_matrix.tsv\"\n",
        "# norm_save_path = f\"{accession}_expression_matrix.tsv\"\n",
        "# tdq.preprocess.normalize(norm_input_path, norm_save_path, clinical_file)\n",
        "\n",
        "# Read normalized expression matrix from file\n",
        "expr_df = pd.read_csv(norm_save_path, sep=\"\\t\", index_col=0)\n",
        "\n",
        "# Clean up sample names (extract \"GSMxxxx\")\n",
        "expr_df = tdq.geo.clean_geo_sample_columns(expr_df)\n",
        "\n",
        "# Save expression matrix with cleaned sample names\n",
        "cleaned_save_path = norm_save_path  # overwrite old file\n",
        "expr_df.to_csv(cleaned_save_path, sep=\"\\t\")\n",
        "\n",
        "# Preview current expression matrix\n",
        "print(f\"Expression matrix for {accession} after normalization:\")\n",
        "expr_df.iloc[:8, :3]"
      ],
      "metadata": {
        "id": "0HXlbINxrvyk",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 355
        },
        "outputId": "2e975e20-be38-4d57-82de-62c2257b1587"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Executing: Rscript /usr/local/lib/python3.10/dist-packages/transcriptomics_data_query/rscripts/rma_normalization.R GSE161750 GSE161750_expression_matrix.tsv\n",
            "Normalization complete.\n",
            "Expression matrix for GSE161750 after normalization:\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "          GSM4913974  GSM4913975  GSM4913976\n",
              "23064070    8.456688    8.332458    8.521047\n",
              "23064071    7.780182    7.467921    7.677162\n",
              "23064072    5.602551    6.618323    6.485193\n",
              "23064073    7.595782    7.153539    7.549301\n",
              "23064074    5.948645    5.597582    5.451984\n",
              "23064075    8.047622    7.734275    8.066836\n",
              "23064076    6.540991    6.105066    6.668104\n",
              "23064077    9.316287    9.071693    9.154063"
            ],
            "text/html": [
              "\n",
              "\n",
              "  <div id=\"df-ae23dd77-3a6a-4ca1-b1bf-a0fb1fec43e3\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GSM4913974</th>\n",
              "      <th>GSM4913975</th>\n",
              "      <th>GSM4913976</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>23064070</th>\n",
              "      <td>8.456688</td>\n",
              "      <td>8.332458</td>\n",
              "      <td>8.521047</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064071</th>\n",
              "      <td>7.780182</td>\n",
              "      <td>7.467921</td>\n",
              "      <td>7.677162</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064072</th>\n",
              "      <td>5.602551</td>\n",
              "      <td>6.618323</td>\n",
              "      <td>6.485193</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064073</th>\n",
              "      <td>7.595782</td>\n",
              "      <td>7.153539</td>\n",
              "      <td>7.549301</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064074</th>\n",
              "      <td>5.948645</td>\n",
              "      <td>5.597582</td>\n",
              "      <td>5.451984</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064075</th>\n",
              "      <td>8.047622</td>\n",
              "      <td>7.734275</td>\n",
              "      <td>8.066836</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064076</th>\n",
              "      <td>6.540991</td>\n",
              "      <td>6.105066</td>\n",
              "      <td>6.668104</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23064077</th>\n",
              "      <td>9.316287</td>\n",
              "      <td>9.071693</td>\n",
              "      <td>9.154063</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ae23dd77-3a6a-4ca1-b1bf-a0fb1fec43e3')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "\n",
              "\n",
              "\n",
              "    <div id=\"df-82734bfb-babb-4ee5-94e4-0f1f6ce36baf\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-82734bfb-babb-4ee5-94e4-0f1f6ce36baf')\"\n",
              "              title=\"Suggest charts.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "    </div>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "    background-color: #E8F0FE;\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: #1967D2;\n",
              "    height: 32px;\n",
              "    padding: 0 0 0 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: #E2EBFA;\n",
              "    box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: #174EA6;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "    background-color: #3B4455;\n",
              "    fill: #D2E3FC;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart:hover {\n",
              "    background-color: #434B5C;\n",
              "    box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "    filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "    fill: #FFFFFF;\n",
              "  }\n",
              "</style>\n",
              "\n",
              "    <script>\n",
              "      async function quickchart(key) {\n",
              "        const containerElement = document.querySelector('#' + key);\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      }\n",
              "    </script>\n",
              "\n",
              "      <script>\n",
              "\n",
              "function displayQuickchartButton(domScope) {\n",
              "  let quickchartButtonEl =\n",
              "    domScope.querySelector('#df-82734bfb-babb-4ee5-94e4-0f1f6ce36baf button.colab-df-quickchart');\n",
              "  quickchartButtonEl.style.display =\n",
              "    google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "}\n",
              "\n",
              "        displayQuickchartButton(document);\n",
              "      </script>\n",
              "      <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-ae23dd77-3a6a-4ca1-b1bf-a0fb1fec43e3 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-ae23dd77-3a6a-4ca1-b1bf-a0fb1fec43e3');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### 5b Map probes to genes (for microarray data only)\n",
        "expr_df = tdq.geo.map_probes_to_genes(expr_df, gse)\n",
        "\n",
        "# Overwrite previous expression matrix file\n",
        "expr_df.to_csv(cleaned_save_path, sep=\"\\t\")\n",
        "\n",
        "# Preview current expression matrix\n",
        "print(f\"Expression matrix for {accession} after mapping probes to genes:\")\n",
        "expr_df.iloc[:8, :3]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 557
        },
        "id": "MeqPmuAe6X1B",
        "outputId": "336ce6cc-f576-4489-8c4d-590ebfcd1a78"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "15-Aug-2023 02:52:57 DEBUG utils - Directory ./ already exists. Skipping.\n",
            "DEBUG:GEOparse:Directory ./ already exists. Skipping.\n",
            "15-Aug-2023 02:52:57 INFO GEOparse - File already exist: using local version.\n",
            "INFO:GEOparse:File already exist: using local version.\n",
            "15-Aug-2023 02:52:57 INFO GEOparse - Parsing ./GPL23159.txt: \n",
            "INFO:GEOparse:Parsing ./GPL23159.txt: \n",
            "15-Aug-2023 02:52:57 DEBUG GEOparse - PLATFORM: GPL23159\n",
            "DEBUG:GEOparse:PLATFORM: GPL23159\n",
            "15-Aug-2023 02:53:00 WARNING GEOTypes - Detected duplicated columns in d GPL23159. Correcting.\n",
            "\n",
            "WARNING:GEOparse:Detected duplicated columns in d GPL23159. Correcting.\n",
            "\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Using annotation column SPOT_ID.1 for Ensembl IDs\n",
            "Expression matrix for GSE161750 after mapping probes to genes:\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                 GSM4913974  GSM4913975  GSM4913976\n",
              "ENSG00000093100    8.336460    7.974318    8.300854\n",
              "ENSG00000106540    6.611486    6.178451    6.286634\n",
              "ENSG00000137808    5.914954    5.773396    5.592572\n",
              "ENSG00000145063    6.823582    6.514701    7.082004\n",
              "ENSG00000151303    7.108568    7.324181    7.424647\n",
              "ENSG00000157306    8.341304    8.440072    8.573010\n",
              "ENSG00000167355    4.098994    4.081819    4.494833\n",
              "ENSG00000170089    7.708660    8.472836    8.133657"
            ],
            "text/html": [
              "\n",
              "\n",
              "  <div id=\"df-1f3ff42d-1d78-407a-9ed7-a161da2377f5\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GSM4913974</th>\n",
              "      <th>GSM4913975</th>\n",
              "      <th>GSM4913976</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>ENSG00000093100</th>\n",
              "      <td>8.336460</td>\n",
              "      <td>7.974318</td>\n",
              "      <td>8.300854</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000106540</th>\n",
              "      <td>6.611486</td>\n",
              "      <td>6.178451</td>\n",
              "      <td>6.286634</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000137808</th>\n",
              "      <td>5.914954</td>\n",
              "      <td>5.773396</td>\n",
              "      <td>5.592572</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000145063</th>\n",
              "      <td>6.823582</td>\n",
              "      <td>6.514701</td>\n",
              "      <td>7.082004</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000151303</th>\n",
              "      <td>7.108568</td>\n",
              "      <td>7.324181</td>\n",
              "      <td>7.424647</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000157306</th>\n",
              "      <td>8.341304</td>\n",
              "      <td>8.440072</td>\n",
              "      <td>8.573010</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000167355</th>\n",
              "      <td>4.098994</td>\n",
              "      <td>4.081819</td>\n",
              "      <td>4.494833</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENSG00000170089</th>\n",
              "      <td>7.708660</td>\n",
              "      <td>8.472836</td>\n",
              "      <td>8.133657</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1f3ff42d-1d78-407a-9ed7-a161da2377f5')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "\n",
              "\n",
              "\n",
              "    <div id=\"df-4603a597-56ae-4e63-a187-9ccac7e8f347\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-4603a597-56ae-4e63-a187-9ccac7e8f347')\"\n",
              "              title=\"Suggest charts.\"\n",
              "              style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "    </div>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "    background-color: #E8F0FE;\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: #1967D2;\n",
              "    height: 32px;\n",
              "    padding: 0 0 0 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: #E2EBFA;\n",
              "    box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: #174EA6;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "    background-color: #3B4455;\n",
              "    fill: #D2E3FC;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart:hover {\n",
              "    background-color: #434B5C;\n",
              "    box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "    filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "    fill: #FFFFFF;\n",
              "  }\n",
              "</style>\n",
              "\n",
              "    <script>\n",
              "      async function quickchart(key) {\n",
              "        const containerElement = document.querySelector('#' + key);\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      }\n",
              "    </script>\n",
              "\n",
              "      <script>\n",
              "\n",
              "function displayQuickchartButton(domScope) {\n",
              "  let quickchartButtonEl =\n",
              "    domScope.querySelector('#df-4603a597-56ae-4e63-a187-9ccac7e8f347 button.colab-df-quickchart');\n",
              "  quickchartButtonEl.style.display =\n",
              "    google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "}\n",
              "\n",
              "        displayQuickchartButton(document);\n",
              "      </script>\n",
              "      <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-1f3ff42d-1d78-407a-9ed7-a161da2377f5 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-1f3ff42d-1d78-407a-9ed7-a161da2377f5');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### 6. Further processing\n",
        "\n",
        "# The annotation table for GSE161750 (platform GPL23159) contains Ensembl IDs.\n",
        "# Suppose we are interested in human matrisome gene expression, and want them as gene symbols.\n",
        "# We can use the preprocess module to obtain a matrisome expression matrix and convert to symbols."
      ],
      "metadata": {
        "id": "7MxvjHDe9HZz"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### 6.1. Get gene symbols in the KEGG pathway for olfactory transduction\n",
        "gene_set = \"KEGG_OLFACTORY_TRANSDUCTION\"\n",
        "gene_set_symbols = tdq.preprocess.get_genes_from_msig_set(gene_set)\n",
        "# Alternatively you can read genes from a text file (1 gene per line):\n",
        "# gene_set_symbols = tdq.preprocess.get_genes_from_file(\"genes_of_interest.txt\")\n",
        "\n",
        "# Preview gene symbols in the list\n",
        "gene_set_symbols[:10]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "apfaqlt7-Bo-",
        "outputId": "819783ac-2957-473f-c1c4-ad59469540f3"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['ADCY3',\n",
              " 'ARRB2',\n",
              " 'CALM1',\n",
              " 'CALM2',\n",
              " 'CALM3',\n",
              " 'CALML3',\n",
              " 'CALML5',\n",
              " 'CALML6',\n",
              " 'CAMK2A',\n",
              " 'CAMK2B']"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 6.2. Obtain the ensembl IDs for this set of genes\n",
        "# The convert_genes function uses mygene.info to convert genes from one format to another.\n",
        "# The function accepts 3 formats: symbol, Enseml ID (ensembl.gene), and Entrez ID (entrezgene).\n",
        "\n",
        "# Convert symbols to Ensembl IDs\n",
        "gene_set_ensembl = tdq.preprocess.convert_genes(\n",
        "    genes=gene_set_symbols, in_format=\"symbol\", out_format=\"ensembl.gene\", species=\"human\")\n",
        "\n",
        "# Discard not found genes\n",
        "gene_set_ensembl.dropna(inplace=True)\n",
        "\n",
        "# Preview Ensembl IDs in the gene set\n",
        "gene_set_ensembl.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VeuO8rSK-C-p",
        "outputId": "72887d26-aad4-4c3a-f7a8-d1cbcbccd488"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:biothings.client:1 input query terms found dup hits:\t[('OR10J3', 2)]\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "query\n",
              "ADCY3    ENSG00000138031\n",
              "ARRB2    ENSG00000141480\n",
              "CALM1    ENSG00000198668\n",
              "CALM2    ENSG00000143933\n",
              "CALM3    ENSG00000160014\n",
              "Name: ensembl.gene, dtype: object"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 6.3. Select rows in `expr_df` that contain these genes\n",
        "filtered_expression = tdq.preprocess.select_rows(expr_df, gene_set_ensembl)\n",
        "\n",
        "# 6.5. Preview the filtered expression matrix\n",
        "print(f\"\\nFiltered expression matrix for {accession} (Ensembl IDs)\")\n",
        "print(filtered_expression.iloc[:8, :3])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jwMaWuq59-nJ",
        "outputId": "c297a47e-2ff5-49bd-b01c-83a82a92894d"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Filtered expression matrix for GSE161750 (Ensembl IDs)\n",
            "                 GSM4913974  GSM4913975  GSM4913976\n",
            "ENSG00000167355    4.098994    4.081819    4.494833\n",
            "ENSG00000181214    4.359115    4.267042    4.459215\n",
            "ENSG00000278870    3.904819    3.802722    4.531649\n",
            "ENSG00000279051    5.921605    5.524552    5.805795\n",
            "ENSG00000279263    3.919104    3.831666    4.168566\n",
            "ENSG00000279301    5.101743    4.960626    4.930736\n",
            "ENSG00000279761    3.926455    3.914708    3.891808\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 6.4. Optionally, convert the Ensemble IDs to gene symbols\n",
        "genes = filtered_expression.index\n",
        "filtered_expression.index = tdq.preprocess.convert_genes(genes, \"ensembl.gene\", \"symbol\")\n",
        "\n",
        "# Preview the matrisome expression matrix\n",
        "print(f\"\\nFiltered expression matrix for {accession} (gene symbols)\")\n",
        "print(filtered_expression.iloc[:8, :3])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UkiYbU7K-k-1",
        "outputId": "2b5bd717-9d16-4e86-bccc-971669d036de"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Filtered expression matrix for GSE161750 (gene symbols)\n",
            "        GSM4913974  GSM4913975  GSM4913976\n",
            "symbol                                    \n",
            "OR51B5    4.098994    4.081819    4.494833\n",
            "OR8G2P    4.359115    4.267042    4.459215\n",
            "OR51G1    3.904819    3.802722    4.531649\n",
            "OR6Q1     5.921605    5.524552    5.805795\n",
            "OR2L8     3.919104    3.831666    4.168566\n",
            "OR2T11    5.101743    4.960626    4.930736\n",
            "OR5D13    3.926455    3.914708    3.891808\n"
          ]
        }
      ]
    }
  ]
}